--- a/Core/HW/MediaEngine.cpp	2018-10-26 20:12:11.573955557 +0000
+++ b/Core/HW/MediaEngine.cpp	2018-10-26 20:12:18.855023903 +0000
@@ -316,7 +316,7 @@
 	}
 	av_dict_free(&open_opt);
 
-	if (!SetupStreams()) {
+	if (true) {
 		// Fallback to old behavior.  Reads too much and corrupts when game doesn't read fast enough.
 		// SetupStreams sometimes work for newer FFmpeg 3.1+ now, but sometimes framerate is missing.
 		WARN_LOG_REPORT_ONCE(setupStreams, ME, "Failed to read valid video stream data from header");
