--- a/kernel/nv-drm.c	2023-10-22 05:50:14.478811549 +0000
+++ b/kernel/nv-drm.c	2023-10-22 06:00:35.172438953 +0000
@@ -509,7 +509,9 @@
     .gem_free_object_unlocked = nv_gem_free,
 #endif
 
+#if LINUX_VERSION_CODE <  KERNEL_VERSION(6, 6, 0)
     .prime_handle_to_fd = drm_gem_prime_handle_to_fd,
+#endif
 #if LINUX_VERSION_CODE <= KERNEL_VERSION(5, 10, 0)
     .gem_prime_export = drm_gem_prime_export,
     .gem_prime_get_sg_table = nv_gem_prime_get_sg_table,
