--- a/src/basic/check-filesystems.sh	2022-02-01 21:47:06.181142404 +0100
+++ b/src/basic/check-filesystems.sh	2022-02-01 21:45:58.760989671 +0100
@@ -23,7 +23,8 @@ for fs in $($cpp -dM $includes - </dev/n
         # STACK_END_MAGIC doesn't refer to a filesystem
         # mtd_inode was removed in 2015
         # futexfs was removed in 2018
-        if [[ "$fs" =~ ^(STACK_END_MAGIC|MTD_INODE_FS_MAGIC|FUTEXFS_SUPER_MAGIC)$ ]]; then
+        # sdcardfs is Android thing
+        if [[ "$fs" =~ ^(STACK_END_MAGIC|MTD_INODE_FS_MAGIC|FUTEXFS_SUPER_MAGIC|SDCARDFS_SUPER_MAGIC)$ ]]; then
             continue
         fi
         echo "Filesystem found in kernel header but not in $(basename "$filesystems_gperf"): $fs";
