From 4ee9769b5746240247c3a01f4000afdf8fc56857 Mon Sep 17 00:00:00 2001
From: joed74 <gh@dolze.de>
Date: Tue, 10 Sep 2024 15:57:59 +0200
Subject: [PATCH] Fixed import for vdr 2.7.1

---
 import.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/import.cpp b/import.cpp
index 1bb1849..8c73024 100644
--- a/import.cpp
+++ b/import.cpp
@@ -86,7 +86,11 @@ cEvent *cImport::SearchVDREventByTitle(cEPGSource *source, cSchedule* schedule,
     const char *cxTitle=conv->Convert(Title);
 
     // 2nd with StartTime
+#if VDRVERSNUM >= 20701
+    cEvent *f=(cEvent *) schedule->GetEventByTime(StartTime+hint);
+#else
     cEvent *f=(cEvent *) schedule->GetEvent((tEventID) 0,StartTime+hint);
+#endif
     if (f)
     {
         if (!strcasecmp(f->Title(),cxTitle))
@@ -188,10 +192,18 @@ cEvent *cImport::SearchVDREvent(cEPGSource *source, cSchedule* schedule, cXMLTVE
 
     // try to find an event,
     // 1st with our own EventID
+#if VDRVERSNUM >= 20701
+    if (xevent->EITEventID()) f=(cEvent *) schedule->GetEventById(xevent->EITEventID());
+#else
     if (xevent->EITEventID()) f=(cEvent *) schedule->GetEvent(xevent->EITEventID());
+#endif
     if (f) return f;
 
+#if VDRVERSNUM >= 20701
+    if (xevent->EventID() && append) f=(cEvent *) schedule->GetEventById(xevent->EITEventID());
+#else
     if (xevent->EventID() && append) f=(cEvent *) schedule->GetEvent(xevent->EventID());
+#endif
     if (f) return f;
 
     f=SearchVDREventByTitle(source, schedule, xevent->Title(), xevent->StartTime(),
