From 8286d585d6177c3c3a0321459bfaa9e1e5d8122c Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Fri, 1 Dec 2023 12:34:41 +0000
Subject: [PATCH] use const xmlErrorPtr to compile with libxml2.12

---
 parse.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parse.cpp b/parse.cpp
index a2ab767..87d8f16 100644
--- a/parse.cpp
+++ b/parse.cpp
@@ -1070,7 +1070,7 @@ int cParse::Process(cEPGExecutor &myExecutor,char *buffer, int bufsize)
             skipped++;
             continue;
         }
-        xmlErrorPtr xmlerr=xmlGetLastError();
+        const xmlError* xmlerr=xmlGetLastError();
         if (xmlerr && xmlerr->code)
         {
             esyslogs(source,"%s",xmlerr->message);
-- 
2.34.1

