--- vdr-plugin-xmltv2vdr-0.2.2/Makefile	2023-09-29 18:10:08.000000000 +0000
+++ vdr-plugin-xmltv2vdr-0.2.2/Makefile	2023-11-08 14:02:23.031530482 +0000
@@ -53,6 +53,8 @@
 PKG-LIBS += libxml-2.0 libpcrecpp sqlite3
 PKG-INCLUDES += libxml-2.0 libpcrecpp sqlite3
 
+INCLUDES += -I$(VDRDIR)/include
+
 DEFINES += -D_GNU_SOURCE -D_XOPEN_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
 
 CXXFLAGS += $(shell $(PKG_CONFIG) --cflags $(PKG-INCLUDES))
