--- a/configure
+++ b/configure
@@ -12000,7 +12000,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-	if test "$cross_compiling" = yes; then
+	if test "$cross_compiling" != yes; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -12061,7 +12061,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-	if test "$cross_compiling" = yes; then
+	if test "$cross_compiling" != yes; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
@@ -12137,7 +12137,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
 
-	if test "$cross_compiling" = yes; then
+	if test "$cross_compiling" != yes; then
   { { echo "$as_me:$LINENO: error: cannot run test program while cross compiling
 See \`config.log' for more details." >&5
 echo "$as_me: error: cannot run test program while cross compiling
