From 6cadc2e38b1cefe879fb97fd192e39cb102c4d3c Mon Sep 17 00:00:00 2001
From: Radostan Riedel <raybuntu@gmail.com>
Date: Wed, 18 Oct 2017 19:42:55 +0000
Subject: [PATCH] skip shlibsign

diff --git a/nss/cmd/shlibsign/Makefile b/nss/cmd/shlibsign/Makefile
--- a/nss/cmd/shlibsign/Makefile
+++ b/nss/cmd/shlibsign/Makefile
@@ -100,7 +100,9 @@ SRCDIR = $(call core_abspath,.)
     endif
 
 
+ifndef SKIP_SHLIBSIGN
 libs: install
 ifdef CHECKLOC
 	$(MAKE) $(CHECKLOC)
 endif
+endif
