From 6cadc2e38b1cefe879fb97fd192e39cb102c4d3c Mon Sep 17 00:00:00 2001
From: Radostan Riedel <raybuntu@gmail.com>
Date: Wed, 18 Oct 2017 19:42:55 +0000
Subject: [PATCH] disable host cflags

diff --git a/nspr/config/autoconf.mk.in b/nspr/config/autoconf.mk.in
index ba95532..f0038f9 100644
--- a/nspr/config/autoconf.mk.in
+++ b/nspr/config/autoconf.mk.in
@@ -103,7 +103,7 @@ DSO_LDOPTS	= @DSO_LDOPTS@
 RESOLVE_LINK_SYMBOLS = @RESOLVE_LINK_SYMBOLS@
 
 HOST_CC		= @HOST_CC@
-HOST_CFLAGS	= @HOST_CFLAGS@
+#HOST_CFLAGS	= @HOST_CFLAGS@
 HOST_LDFLAGS	= @HOST_LDFLAGS@
 
 DEFINES		= @DEFINES@ @DEFS@
-- 
2.53.0

