From d75640278574196982daaed7e6bd917caee4f91a Mon Sep 17 00:00:00 2001
From: cdu13a <cdu13a@gmail.com>
Date: Sat, 9 Nov 2019 07:38:53 -0500
Subject: add Device Id's

---
 os_dep/linux/usb_intf.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/os_dep/linux/usb_intf.c b/os_dep/linux/usb_intf.c
index ada1c07..7dda69f 100644
--- a/os_dep/linux/usb_intf.c
+++ b/os_dep/linux/usb_intf.c
@@ -201,6 +201,8 @@ static struct usb_device_id rtw_usb_id_tbl[] = {
 	{USB_DEVICE(0x0411, 0x0242), .driver_info = RTL8821}, /* ELECOM - WDC-433DU2H */
 	{USB_DEVICE(0x2019, 0xAB32), .driver_info = RTL8821}, /* Planex - GW-450S */
 	{USB_DEVICE(0x0846, 0x9052), .driver_info = RTL8821}, /* Netgear - A6100 */
+	{USB_DEVICE(0x2357, 0x011e), .driver_info = RTL8821}, /* TP-Link Archer T2U Nano */
+	{USB_DEVICE(0x2357, 0x011f), .driver_info = RTL8821}, /* TP-Link Archer T2U AC600 */
 	/*=== Patched ID ===*/
 	{USB_DEVICE(0x056e, 0x4007),.driver_info = RTL8821}, /* Elecom - WDC-433DU2HBK */
 	{USB_DEVICE(0x0BDA, 0xA811),.driver_info = RTL8821}, /* Comfast - CF-915AC */	
-- 
2.17.1

