From 13949bec52ca1a043c2a3ba6aeb0442d33122751 Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Fri, 8 Feb 2019 17:01:01 +0000
Subject: [PATCH] build native-tools with host compiler not target

---
 CMakeLists.txt | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 39b8025..7388370 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1282,11 +1282,15 @@ ELSE()
 ENDIF()
 
 SET(NATIVE_SRC_DIR  "${OpenAL_SOURCE_DIR}/native-tools/")
-SET(NATIVE_BIN_DIR  "${OpenAL_BINARY_DIR}/native-tools/")
+if(NOT NATIVE_BIN_DIR)
+  SET(NATIVE_BIN_DIR  "${OpenAL_BINARY_DIR}/native-tools/")
+endif()
 FILE(MAKE_DIRECTORY "${NATIVE_BIN_DIR}")
 
+option(ALSOFT_NATIVE_TOOLS "Build native tools for host" ON)
 SET(BIN2H_COMMAND  "${NATIVE_BIN_DIR}bin2h")
 SET(BSINCGEN_COMMAND  "${NATIVE_BIN_DIR}bsincgen")
+if(ALSOFT_NATIVE_TOOLS)
 ADD_CUSTOM_COMMAND(OUTPUT "${BIN2H_COMMAND}" "${BSINCGEN_COMMAND}"
     COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" "${NATIVE_SRC_DIR}"
     COMMAND ${CMAKE_COMMAND} -E remove "${BIN2H_COMMAND}" "${BSINCGEN_COMMAND}"
@@ -1297,6 +1301,7 @@ ADD_CUSTOM_COMMAND(OUTPUT "${BIN2H_COMMAND}" "${BSINCGEN_COMMAND}"
                      C "${NATIVE_SRC_DIR}bsincgen.c"
     VERBATIM
 )
+endif()
 ADD_CUSTOM_TARGET(native-tools
     DEPENDS "${BIN2H_COMMAND}" "${BSINCGEN_COMMAND}"
     VERBATIM
-- 
2.7.4
