From a12648520ae79d44bd056a22e6a3173ecb45ffa9 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Mon, 25 Apr 2016 10:37:14 -0700
Subject: [PATCH] makefile

diff --git a/Makefile b/Makefile
index 130afd8..480032e 100644
--- a/Makefile
+++ b/Makefile
@@ -43,8 +43,8 @@ clean:
 	rm -f $(OBJS) $(TARGET)
 
 install: $(TARGET)
-	install -D -g root -o root $(TARGET) $(TARGET_DIR)/sbin/$(TARGET)
-	install -D -g root -o root $(TARGET).1 $(TARGET_DIR)/share/man/man1/$(TARGET).1
+	install -D $(TARGET) $(TARGET_DIR)/sbin/$(TARGET)
+	install -D $(TARGET).1 $(TARGET_DIR)/share/man/man1/$(TARGET).1
 
 hd-idle.o:     hd-idle.c
 
