From 517ed274d6025e44f4637af198a1b37fdeff85ed Mon Sep 17 00:00:00 2001
From: heitbaum <rudi@heitbaum.com>
Date: Sun, 27 Dec 2020 02:58:08 +0000
Subject: [PATCH] disable docs

diff --git a/doc/Makefile.am b/doc/Makefile.am
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,7 +1,7 @@
 # AM_MAKEFLAGS = @MAKEFLAGS@
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = www man
+SUBDIRS =
 
 if DOXYGEN
 all-local: html
