From 28cd331e5f9abeacf7028a933879e1f3e6af9c9c Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Sun, 10 Jan 2016 00:17:59 +0100
Subject: [PATCH] use pkgconfig

diff --git a/m4/md_check_libusb018b.m4 b/m4/md_check_libusb018b.m4
--- a/m4/md_check_libusb018b.m4
+++ b/m4/md_check_libusb018b.m4
@@ -2,8 +2,8 @@
   [
     AC_CHECK_HEADERS([usb.h])
 
-    LIBUSB_CFLAGS="`libusb-config --cflags`"
-    LIBUSB_LIBS="`libusb-config --libs`"
+    LIBUSB_CFLAGS="`pkg-config --cflags libusb`"
+    LIBUSB_LIBS="`pkg-config --libs libusb`"
 
     AC_SUBST(LIBUSB_CFLAGS)
     AC_SUBST(LIBUSB_LIBS)
