From c3ecb068b9d649fa3c6bf66ced398c4acb3c7078 Mon Sep 17 00:00:00 2001
From: Portisch <hugo.portisch@yahoo.de>
Date: Thu, 8 Jul 2021 12:57:22 +0200
Subject: [PATCH 1/2] cpuinfo: add S905Y2 rev. b

---
 arch/arm64/kernel/cpuinfo.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/kernel/cpuinfo.c b/arch/arm64/kernel/cpuinfo.c
index ce87ee04f1ec..8453662dbd94 100644
--- a/arch/arm64/kernel/cpuinfo.c
+++ b/arch/arm64/kernel/cpuinfo.c
@@ -138,6 +138,7 @@ static int c_show(struct seq_file *m, void *v)
 		{ "A113D", MESON_CPU_MAJOR_ID_AXG, 0x22, 0xff },
 		{ "S905D2", MESON_CPU_MAJOR_ID_G12A, 0x10, 0xf0 },
 		{ "S905X2", MESON_CPU_MAJOR_ID_G12A, 0x40, 0xf0 },
+		{ "S905Y2", MESON_CPU_MAJOR_ID_G12A, 0x30, 0xf0 },
 		{ "S922X", MESON_CPU_MAJOR_ID_G12B, 0x40, 0xf0 },
 		{ "A311D", MESON_CPU_MAJOR_ID_G12B, 0x10, 0xf0 },
 		{ "S905X3", MESON_CPU_MAJOR_ID_SM1, 0x01, 0x0f },
-- 
2.32.0

