From e6169384f54b885a726643091f3b20310b27d9c8 Mon Sep 17 00:00:00 2001
From: heitbaum <rudi@heitbaum.com>
Date: Sun, 05 Dec 2021 08:20:11 +0000
Subject: [PATCH] add dependant static libraries

diff --git a/src/unwind/libunwind.pc.in b/src/unwind/libunwind.pc.in
--- a/src/unwind/libunwind.pc.in
+++ b/src/unwind/libunwind.pc.in
@@ -6,6 +6,6 @@
 Name: libunwind
 Description: libunwind base library
 Version: @VERSION@
-Libs: -L${libdir} -lunwind
+Libs: -L${libdir} -lunwind @LIBLZMA@ @LIBZ@
 Libs.private: @LIBLZMA@ @LIBZ@
 Cflags: -I${includedir}
