diff -aurNp a/systemd/nfs-server.service b/systemd/nfs-server.service
--- a/systemd/nfs-server.service	2020-02-09 20:41:19.000000000 +0100
+++ b/systemd/nfs-server.service	2020-08-03 11:43:44.697756700 +0200
@@ -19,6 +19,8 @@ Before= rpc-statd-notify.service
 Wants=auth-rpcgss-module.service rpc-svcgssd.service
 After=rpc-gssd.service gssproxy.service rpc-svcgssd.service
 
+ConditionPathExists=/storage/.config/exports
+
 [Service]
 Type=oneshot
 RemainAfterExit=yes
