From c7a9307565c76fd793a6ee87b0442d37173cc087 Mon Sep 17 00:00:00 2001
From: Kai Sommerfeld <kai.sommerfeld@gmx.com>
Date: Sat, 1 Jun 2019 13:23:22 +0200
Subject: [PATCH] [PVR] Fix deadlock on PVR shutdown.

---
 xbmc/pvr/PVRManager.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/xbmc/pvr/PVRManager.cpp b/xbmc/pvr/PVRManager.cpp
index 6b9ea7ae2b3b..d61a8ec99fef 100644
--- a/xbmc/pvr/PVRManager.cpp
+++ b/xbmc/pvr/PVRManager.cpp
@@ -299,8 +299,6 @@ void CPVRManager::Start()
   // lock here (m_startStopMutex), which only gets hold while starting/restarting pvr manager.
   Stop();
 
-  CSingleLock lock(m_critSection);
-
   if (!m_addons->HasCreatedClients())
     return;
 
@@ -339,8 +337,6 @@ void CPVRManager::Stop(void)
 
   SetState(ManagerStateInterrupted);
 
-  CSingleLock lock(m_critSection);
-
   UnloadComponents();
   m_database->Close();
 
From eb5c56891ef4739566898b9bc72914a9c8ba8be8 Mon Sep 17 00:00:00 2001
From: Arthur Liberman <arthur_liberman@hotmail.com>
Date: Fri, 31 May 2019 01:50:13 +0300
Subject: [PATCH] settings: add "Dark Grey" subtitles color option This can be
 useful when watching HDR content with subs, and "Grey" is still too bright.

---
 addons/resource.language.en_gb/resources/strings.po       | 8 ++++++--
 system/settings/settings.xml                              | 1 +
 .../VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecTX3G.cpp | 2 +-
 .../VideoPlayer/VideoRenderers/OverlayRendererGUI.cpp     | 5 +++--
 xbmc/utils/Color.h                                        | 1 +
 5 files changed, 12 insertions(+), 5 deletions(-)

diff --git a/addons/resource.language.en_gb/resources/strings.po b/addons/resource.language.en_gb/resources/strings.po
index 74c74a9c8353..eda5521614a1 100644
--- a/addons/resource.language.en_gb/resources/strings.po
+++ b/addons/resource.language.en_gb/resources/strings.po
@@ -3458,8 +3458,12 @@ msgctxt "#767"
 msgid "Grey"
 msgstr ""
 
-#empty strings from id 768 to 769
-#strings 768 and 769 reserved for more subtitle colours
+msgctxt "#768"
+msgid "Dark Grey"
+msgstr ""
+
+#empty string id 769
+#string 769 reserved for more subtitle colours
 
 msgctxt "#770"
 msgid "Error {0:d}: share not available"
diff --git a/system/settings/settings.xml b/system/settings/settings.xml
index 5a136982058b..5f858f1a27ec 100755
--- a/system/settings/settings.xml
+++ b/system/settings/settings.xml
@@ -553,6 +553,7 @@
               <option label="765">5</option> <!-- Cyan -->
               <option label="766">6</option> <!-- Light grey -->
               <option label="767">7</option> <!-- Grey -->
+              <option label="768">8</option> <!-- Dark Grey -->
             </options>
           </constraints>
           <dependencies>
diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecTX3G.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecTX3G.cpp
index 15f87cbbaf07..c89b7bfd8520 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecTX3G.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Overlay/DVDOverlayCodecTX3G.cpp
@@ -62,7 +62,7 @@ CDVDOverlayCodecTX3G::CDVDOverlayCodecTX3G() : CDVDOverlayCodec("TX3G Subtitle D
 {
   m_pOverlay = NULL;
   // stupid, this comes from a static global in GUIWindowFullScreen.cpp
-  uint32_t colormap[8] = { 0xFFFFFF00, 0xFFFFFFFF, 0xFF0099FF, 0xFF00FF00, 0xFFCCFF00, 0xFF00FFFF, 0xFFE5E5E5, 0xFFC0C0C0 };
+  uint32_t colormap[9] = { 0xFFFFFF00, 0xFFFFFFFF, 0xFF0099FF, 0xFF00FF00, 0xFFCCFF00, 0xFF00FFFF, 0xFFE5E5E5, 0xFFC0C0C0, 0xFF808080 };
   m_textColor = colormap[CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_SUBTITLES_COLOR)];
 }
 
diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGUI.cpp b/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGUI.cpp
index 62da9b32e87b..47a18fb16d1a 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGUI.cpp
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGUI.cpp
@@ -25,14 +25,15 @@
 
 using namespace OVERLAY;
 
-static UTILS::Color colors[8] = { UTILS::COLOR::YELLOW,
+static UTILS::Color colors[9] = { UTILS::COLOR::YELLOW,
                                   UTILS::COLOR::WHITE,
                                   UTILS::COLOR::BLUE,
                                   UTILS::COLOR::BRIGHTGREEN,
                                   UTILS::COLOR::YELLOWGREEN,
                                   UTILS::COLOR::CYAN,
                                   UTILS::COLOR::LIGHTGREY,
-                                  UTILS::COLOR::GREY };
+                                  UTILS::COLOR::GREY,
+                                  UTILS::COLOR::DARKGREY };
 
 CGUITextLayout* COverlayText::GetFontLayout(const std::string &font, int color, int opacity, int height, int style,
                                             const std::string &fontcache, const std::string &fontbordercache)
diff --git a/xbmc/utils/Color.h b/xbmc/utils/Color.h
index 8c813517f238..5036ccd2cbee 100644
--- a/xbmc/utils/Color.h
+++ b/xbmc/utils/Color.h
@@ -27,5 +27,6 @@ namespace COLOR
   static const Color BRIGHTGREEN = 0xFF00FF00;
   static const Color YELLOWGREEN = 0xFFCCFF00;
   static const Color CYAN = 0xFF00FFFF;
+  static const Color DARKGREY = 0xFF808080;
 } // namespace COLOR
 } // namespace UTILS
