Fix freeing not allocated fallback entries by allocating a copy.

--- a/ncurses/tinfo/tinfo_driver.c
+++ b/ncurses/tinfo/tinfo_driver.c
@@ -181,6 +181,8 @@ drv_CanHandle(TERMINAL_CONTROL_BLOCK * T
     if (status != TGETENT_YES) {
 	const TERMTYPE2 *fallback = _nc_fallback2(tname);
 
+	if (fallback)
+            fallback = _nc_copy_entry(fallback);
 	if (fallback) {
 	    T(("found fallback entry"));
 	    TerminalType(termp) = *fallback;
