--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -100,7 +100,7 @@ stage1scan.c: scan.l stage1flex$(EXEEXT)
 	./stage1flex$(EXEEXT) $(AM_LFLAGS) $(LFLAGS) -o $@ $(srcdir)/scan.l
 else
 stage1scan.c: scan.c
-	sed 's|^\(#line .*\)"'`printf %s $< | sed 's|[][\\\\.*]|\\\\&|g'`'"|\1"$@"|g' $< > $@
+	../../.$(HOST_NAME)/src/stage1flex$(EXEEXT) $(AM_LFLAGS) $(LFLAGS) -o $@ $(srcdir)/scan.l
 endif
 
 dist-hook: scan.l flex$(EXEEXT)
