From 6b4ff6cb9cae1e206cb63d4541a8a272c5bc6535 Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Wed, 08 Jan 2020 13:00:45 +0100
Subject: [PATCH] nocrap

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -43,10 +43,6 @@ EXTRA_DIST = \
 
 SUBDIRS = \
 	src \
-	doc \
-	examples \
-	po \
-	tests \
 	tools
 
 # Create the ChangeLog, but only if we're inside a git working directory
