diff --git a/src/game_main.cpp b/src/game_main.cpp
index 4be806c..cb89232 100644
--- a/src/game_main.cpp
+++ b/src/game_main.cpp
@@ -201,6 +201,15 @@ void gameMain(const CommandLineOptions& options) {
     SDL_INIT_VIDEO | SDL_INIT_AUDIO | SDL_INIT_GAMECONTROLLER));
   auto sdlGuard = defer([]() { SDL_Quit(); });
 
+  SDL_version version;
+  SDL_GetVersion(&version);
+
+  if (version.patch < 10) {
+    if (const auto pMappingsFile = SDL_getenv("SDL_GAMECONTROLLERCONFIG_FILE")) {
+      SDL_GameControllerAddMappingsFromFile(pMappingsFile);
+    }
+  }
+
   sdl_utils::check(SDL_GL_LoadLibrary(nullptr));
   platform::setGLAttributes();
 
