From 7920b919173886f21fd8baf6b9b67249a5f7140b Mon Sep 17 00:00:00 2001
From: 5schatten <36100210+5schatten@users.noreply.github.com>
Date: Tue, 8 Jan 2019 17:01:04 +0100
Subject: [PATCH] Add platform for Amlogic

---
 Makefile | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/Makefile b/Makefile
index 3db2cdda..186330f4 100644
--- a/Makefile
+++ b/Makefile
@@ -144,6 +144,13 @@ USE_SDL2 = 1
     HAVE_NEON = 1
     NAME  = amiberry-vero4k
 
+else ifeq ($(PLATFORM),amlogic)
+USE_SDL2 = 1
+    CFLAGS += -march=armv8-a+crc -mtune=cortex-a73.cortex-a53 -mfpu=neon-fp-armv8
+    CPPFLAGS += -DARMV6_ASSEMBLY -D_FILE_OFFSET_BITS=64 -DARMV6T2 -DUSE_ARMNEON -DARM_HAS_DIV -DUSE_SDL2 -DMALI_GPU -DUSE_RENDER_THREAD -DFASTERCYCLES
+    HAVE_NEON = 1
+    NAME  = amiberry
+
 else ifeq ($(PLATFORM),tinker)
 USE_SDL2 = 1
     CFLAGS += -march=armv7-a -mtune=cortex-a17 -mfpu=neon-vfpv4
