diff --git a/drivers/amlogic/media/enhancement/amvecm/amcsc.c b/drivers/amlogic/media/enhancement/amvecm/amcsc.c
index ac9a91d9fa8c..6e32156a51d6 100644
--- a/drivers/amlogic/media/enhancement/amvecm/amcsc.c
+++ b/drivers/amlogic/media/enhancement/amvecm/amcsc.c
@@ -3836,8 +3836,7 @@ uint32_t sink_hdr_support(const struct vinfo_s *vinfo)
 
 	/* when policy == follow sink(0) or force output (2) */
 	/* use force_output */
-	if ((get_force_output() != 0)
-	&& (get_hdr_policy() != 1)) {
+	if ((get_force_output() != 0)) {
 		switch (get_force_output()) {
 		case BT709:
 			break;
