--- a/configure	2016-10-19 00:05:10.219995829 +0200
+++ b/configure	2016-10-19 00:06:41.154100063 +0200
@@ -12945,44 +12945,10 @@
 
         if test x$have_x = xyes; then
             CFLAGS="$CFLAGS $X_CFLAGS"
-            SYS_GL_LIBS="$X_LIBS -lGL"
-        else
-            SYS_GL_LIBS="-lGL"
+            SYS_GL_LIBS="$X_LIBS -lGL -DHAVE_OPENGL"
         fi
         ;;
 esac
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for OpenGL support" >&5
-$as_echo_n "checking for OpenGL support... " >&6; }
-have_opengl=no
-cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
-
- #include "SDL_opengl.h"
-
-int
-main ()
-{
-
- GLuint texture;
-
-  ;
-  return 0;
-}
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"; then :
-
-have_opengl=yes
-
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.$ac_ext
-{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $have_opengl" >&5
-$as_echo "$have_opengl" >&6; }
-if test x$have_opengl = xyes; then
-    CFLAGS="$CFLAGS -DHAVE_OPENGL"
-    GL_LIBS="$SYS_GL_LIBS"
-else
-    GL_LIBS=""
-fi
 
 
 
