--- a/include/commandline/Option.h
+++ b/include/commandline/Option.h
@@ -15,13 +15,13 @@
 public:
 	Option(const QString &name,
 		  const QString &description = QString(),
-		  const QString &valueName = QString::null,
+		  const QString &valueName = QString(),
 		  const QString &defaultValue = QString()
 	);
 
 	Option(const QStringList &names,
 		   const QString &description = QString(),
-		   const QString &valueName = QString::null,
+		   const QString &valueName = QString(),
 		   const QString &defaultValue = QString()
 	);
 
