diff -Naur hyperion.ng-e438bc62a651c1e6a13396e1430f5687a6ef703f.orig/libsrc/jsonserver/JsonClientConnection.h hyperion.ng-e438bc62a651c1e6a13396e1430f5687a6ef703f/libsrc/jsonserver/JsonClientConnection.h
--- hyperion.ng-e438bc62a651c1e6a13396e1430f5687a6ef703f.orig/libsrc/jsonserver/JsonClientConnection.h	2017-12-01 15:58:17.000000000 +0100
+++ hyperion.ng-e438bc62a651c1e6a13396e1430f5687a6ef703f/libsrc/jsonserver/JsonClientConnection.h	2018-10-26 12:51:37.984397612 +0200
@@ -3,6 +3,7 @@
 // Qt includes
 #include <QString>
 #include <QByteArray>
+#include <QJsonObject>
 
 // util includes
 #include <utils/Logger.h>
