diff --git a/include/bonjour/bonjourservicebrowser.h b/include/bonjour/bonjourservicebrowser.h
index 590747f..c189bff 100644
--- a/include/bonjour/bonjourservicebrowser.h
+++ b/include/bonjour/bonjourservicebrowser.h
@@ -30,7 +30,7 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #define BONJOURSERVICEBROWSER_H
 
 #include <QtCore/QObject>
-#include <dns_sd.h>
+#include <avahi-compat-libdns_sd/dns_sd.h>
 #include "bonjour/bonjourrecord.h"
 
 
diff --git a/include/bonjour/bonjourserviceregister.h b/include/bonjour/bonjourserviceregister.h
index 137f97c..3e36ac0 100755
--- a/include/bonjour/bonjourserviceregister.h
+++ b/include/bonjour/bonjourserviceregister.h
@@ -34,7 +34,7 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "bonjourrecord.h"
 class QSocketNotifier;
 
-#include <dns_sd.h>
+#include <avahi-compat-libdns_sd/dns_sd.h>
 
 class BonjourServiceRegister : public QObject
 {
diff --git a/include/bonjour/bonjourserviceresolver.h b/include/bonjour/bonjourserviceresolver.h
index 08ab381..008d0b2 100644
--- a/include/bonjour/bonjourserviceresolver.h
+++ b/include/bonjour/bonjourserviceresolver.h
@@ -32,7 +32,7 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <QtCore/QObject>
 
 
-#include <dns_sd.h>
+#include <avahi-compat-libdns_sd/dns_sd.h>
 
 class QSocketNotifier;
 class QHostInfo;
