diff --git a/libsrc/boblightserver/CMakeLists.txt b/libsrc/boblightserver/CMakeLists.txt
index 1587d4d..859b09a 100644
--- a/libsrc/boblightserver/CMakeLists.txt
+++ b/libsrc/boblightserver/CMakeLists.txt
@@ -10,5 +10,5 @@ add_library(boblightserver ${BoblightServer_SOURCES} )
 target_link_libraries(boblightserver
 	hyperion
 	hyperion-utils
-	${QT_LIBRARIES}
+	${QT_LIBRARIES} pthread pcre16 dl z
 )
diff --git a/libsrc/grabber/amlogic/CMakeLists.txt b/libsrc/grabber/amlogic/CMakeLists.txt
index 40ce2d5..cff16c1 100644
--- a/libsrc/grabber/amlogic/CMakeLists.txt
+++ b/libsrc/grabber/amlogic/CMakeLists.txt
@@ -15,4 +15,4 @@ add_library(amlogic-grabber ${AmlogicSOURCES} )
 
 target_link_libraries(amlogic-grabber
 	hyperion
-	${QT_LIBRARIES})
+	${QT_LIBRARIES} pthread pcre16 dl z)
diff --git a/libsrc/grabber/dispmanx/CMakeLists.txt b/libsrc/grabber/dispmanx/CMakeLists.txt
index 187de70..5a16e68 100644
--- a/libsrc/grabber/dispmanx/CMakeLists.txt
+++ b/libsrc/grabber/dispmanx/CMakeLists.txt
@@ -17,6 +17,6 @@ add_library(dispmanx-grabber ${DispmanxGrabberSOURCES} )
 
 target_link_libraries(dispmanx-grabber
 	hyperion
-	${QT_LIBRARIES}
+	${QT_LIBRARIES} pthread pcre16 dl z
 	${BCM_LIBRARIES}
 )
diff --git a/libsrc/grabber/framebuffer/CMakeLists.txt b/libsrc/grabber/framebuffer/CMakeLists.txt
index af1368f..b31b14e 100644
--- a/libsrc/grabber/framebuffer/CMakeLists.txt
+++ b/libsrc/grabber/framebuffer/CMakeLists.txt
@@ -8,4 +8,4 @@ add_library(framebuffer-grabber ${FramebufferGrabberSOURCES} )
 
 target_link_libraries(framebuffer-grabber
 	hyperion
-	${QT_LIBRARIES})
+	${QT_LIBRARIES} pthread pcre16 dl z)
diff --git a/libsrc/grabber/osx/CMakeLists.txt b/libsrc/grabber/osx/CMakeLists.txt
index 3a5690c..5de6359 100644
--- a/libsrc/grabber/osx/CMakeLists.txt
+++ b/libsrc/grabber/osx/CMakeLists.txt
@@ -8,4 +8,4 @@ add_library(osx-grabber ${OsxGrabberSOURCES} )
 
 target_link_libraries(osx-grabber
 	hyperion
-	${QT_LIBRARIES})
+	${QT_LIBRARIES} pthread pcre16 dl z)
diff --git a/libsrc/grabber/v4l2/CMakeLists.txt b/libsrc/grabber/v4l2/CMakeLists.txt
index fd1fe93..026fbd8 100644
--- a/libsrc/grabber/v4l2/CMakeLists.txt
+++ b/libsrc/grabber/v4l2/CMakeLists.txt
@@ -8,5 +8,5 @@ add_library(v4l2-grabber ${V4L2_SOURCES} )
 
 target_link_libraries(v4l2-grabber
 	hyperion
-	${QT_LIBRARIES}
+	${QT_LIBRARIES} pthread pcre16 dl z
 )
diff --git a/libsrc/hyperion/CMakeLists.txt b/libsrc/hyperion/CMakeLists.txt
index a974d91..88bde11 100644
--- a/libsrc/hyperion/CMakeLists.txt
+++ b/libsrc/hyperion/CMakeLists.txt
@@ -18,5 +18,5 @@ target_link_libraries(hyperion
 	leddevice
 	bonjour
 	effectengine
-	${QT_LIBRARIES}
+	${QT_LIBRARIES} pthread pcre16 dl z libpng.a
 )
diff --git a/libsrc/udplistener/CMakeLists.txt b/libsrc/udplistener/CMakeLists.txt
index ef222d7..3b003c2 100644
--- a/libsrc/udplistener/CMakeLists.txt
+++ b/libsrc/udplistener/CMakeLists.txt
@@ -10,5 +10,5 @@ add_library(udplistener ${UDPListener_SOURCES} )
 target_link_libraries(udplistener
 	hyperion
 	hyperion-utils
-	${QT_LIBRARIES}
+	${QT_LIBRARIES} pthread pcre16 dl z
 )
