From 3b5abdbea5fdd731b6a0f82da59b7ab58b1c4d39 Mon Sep 17 00:00:00 2001
From: fuzzard <fuzzard@kodi.tv>
Date: Mon, 11 Sep 2023 15:15:19 +1000
Subject: [PATCH] [cmake] Fix linux Install.cmake after PR 22112

---
 cmake/scripts/linux/Install.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/scripts/linux/Install.cmake b/cmake/scripts/linux/Install.cmake
index 3f6c7ac6dac01..1564083b7cbe8 100644
--- a/cmake/scripts/linux/Install.cmake
+++ b/cmake/scripts/linux/Install.cmake
@@ -188,7 +188,7 @@ install(FILES ${CMAKE_SOURCE_DIR}/cmake/scripts/common/AddonHelpers.cmake
               ${CMAKE_SOURCE_DIR}/cmake/scripts/common/ArchSetup.cmake
               ${CMAKE_SOURCE_DIR}/cmake/scripts/common/CheckCommits.cmake
               ${CMAKE_SOURCE_DIR}/cmake/scripts/common/CheckTargetPlatform.cmake
-              ${CMAKE_SOURCE_DIR}/cmake/scripts/common/GenerateVersionedFiles.cmake
+              ${CMAKE_SOURCE_DIR}/cmake/scripts/common/GenerateCompileInfo.cmake
               ${CMAKE_SOURCE_DIR}/cmake/scripts/common/GeneratorSetup.cmake
               ${CMAKE_SOURCE_DIR}/cmake/scripts/common/HandleDepends.cmake
               ${CMAKE_SOURCE_DIR}/cmake/scripts/common/Macros.cmake
