--- a/menu/menu_setting.c
+++ b/menu/menu_setting.c
@@ -10856,7 +10856,6 @@
                MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_REINIT);
             }
 
-#ifdef HAVE_ODROIDGO2
             CONFIG_BOOL(
                   list, list_info,
                   &settings->bools.video_ctx_scaling,
@@ -10873,7 +10872,6 @@
                   SD_FLAG_NONE
                   );
             MENU_SETTINGS_LIST_CURRENT_ADD_CMD(list, list_info, CMD_EVENT_REINIT);
-#endif
 
             CONFIG_UINT(
                   list, list_info,
--- a/msg_hash.h
+++ b/msg_hash.h
@@ -1020,9 +1020,7 @@
    MENU_LABEL(VIDEO_RECORD_THREADS),
    MENU_LABEL(VIDEO_SMOOTH),
    MENU_LABEL(VIDEO_CTX_SCALING),
-#ifdef HAVE_ODROIDGO2
    MENU_LABEL(VIDEO_RGA_SCALING),
-#endif
    
    MENU_LABEL(VIDEO_CROP_OVERSCAN),
 

