--- a/frontend/drivers/platform_unix.c
+++ b/frontend/drivers/platform_unix.c
@@ -1334,12 +1334,17 @@
    }
    #endif
 
-   /* Calculate the brightness */
+   char bright[64];
+   sprintf(bright, "/usr/bin/odroidgoa_utils.sh bright %i", value);
+   system(bright);
+   /*
+   // Calculate the brightness 
    value = (value * max_brightness) / 100;
 
    snprintf(svalue, sizeof(svalue), "%d\n", value);
    filestream_write_file("/sys/class/backlight/backlight/brightness",
                          svalue, strlen(svalue));
+*/
 }
 
 #endif
@@ -1973,11 +1978,13 @@
    android_app_destroy(android_app);
 #endif
 
+/*
 #ifdef HAVE_LAKKA
-   /* Reset brightness to maximum */
+   // Reset brightness to maximum 
    if (settings->uints.screen_brightness != DEFAULT_SCREEN_BRIGHTNESS)
       frontend_unix_set_screen_brightness(DEFAULT_SCREEN_BRIGHTNESS);
 #endif
+*/
 }
 
 static void frontend_unix_init(void *data)
@@ -2800,7 +2807,7 @@
 #else
    NULL,                               /* get_lakka_version */
 #endif
-#if defined(HAVE_LAKKA_SWITCH) || (defined(HAVE_LAKKA) && defined(HAVE_ODROIDGO2))
+#if defined(HAVE_LAKKA_SWITCH) || defined(HAVE_LAKKA)
    frontend_unix_set_screen_brightness,/* set_screen_brightness */
 #else 
    NULL,                         /* set_screen_brightness */
