From 541b9953dd44fbb3d46be38d7243f3c812863d2a Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Fri, 12 Jul 2013 20:37:41 +0200
Subject: [PATCH] storagedir

diff --git a/configure.ac b/configure.ac
index b011e9b..07304c6 100644
--- a/configure.ac
+++ b/configure.ac
@@ -480,10 +480,13 @@ else
 fi
 AC_SUBST(PKGLIBEXECDIR, "${pkglibexecdir}")
 
-if (test "$localstatedir" = '${prefix}/var'); then
-	storagedir="${prefix}/var/lib/bluetooth"
-else
-	storagedir="${localstatedir}/lib/bluetooth"
+
+if (test "x$storagedir" = 'x'); then
+	if (test "$localstatedir" = '${prefix}/var'); then
+		storagedir="${prefix}/var/lib/bluetooth"
+	else
+		storagedir="${localstatedir}/lib/bluetooth"
+	fi
 fi
 AC_DEFINE_UNQUOTED(STORAGEDIR, "${storagedir}",
 			[Directory for the storage files])
-- 
2.53.0

