diff --git a/Makefile.obexd b/Makefile.obexd
index b7e9f2d33..9a9e9a0a0 100644
--- a/Makefile.obexd
+++ b/Makefile.obexd
@@ -1,6 +1,4 @@
 # SPDX-License-Identifier: GPL-2.0
-if OBEX
-
 if SYSTEMD
 systemduserunit_DATA += obexd/src/obex.service
 
@@ -40,6 +38,8 @@ obexd_builtin_sources += obexd/plugins/opp.c
 obexd_builtin_modules += ftp
 obexd_builtin_sources += obexd/plugins/ftp.c obexd/plugins/ftp.h
 
+if OBEX
+
 obexd_builtin_modules += irmc
 obexd_builtin_sources += obexd/plugins/irmc.c
 
@@ -56,6 +56,8 @@ obexd_builtin_sources += obexd/plugins/mas.c obexd/src/map_ap.h \
 				obexd/plugins/messages.h \
 				obexd/plugins/messages-dummy.c
 
+endif
+
 obexd_builtin_modules += mns
 obexd_builtin_sources += obexd/client/mns.c obexd/src/map_ap.h \
 				obexd/client/map-event.h
@@ -107,11 +109,6 @@ obexd_src_obexd_CPPFLAGS = $(AM_CPPFLAGS) $(GLIB_CFLAGS) $(DBUS_CFLAGS) \
 				-D_FILE_OFFSET_BITS=64 \
 				-I$(builddir)/obexd/src
 
-else
-obexd-add-service-symlink:
-obexd-remove-service-symlink:
-endif
-
 if OBEX
 if SYSTEMD
 obexd_src_obexd_CPPFLAGS += -DSYSTEMD
