From c9683fa5240594eab7d2f4a685b469ca5ad9e358 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 05 Feb 2023 14:08:47 +0000
Subject: [PATCH] no docs

diff --git a/Makefile b/Makefile
index ddf2046..a813779 100644
--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,7 @@ else
 endif
 
 .PHONY: all
-all: build docs
+all: build
 
 bin:
 	mkdir -p $@
@@ -117,7 +117,7 @@ install.systemd: $(NV_UNIT_FILES)
 
 
 .PHONY: install
-install: install.netavark install.docs install.systemd
+install: install.netavark install.systemd
 
 .PHONY: uninstall
 uninstall:
-- 
2.53.0

