From 3802cb7b283b50ea387a7b0c41abb43fb86508c6 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 07 Aug 2022 10:54:35 +0000
Subject: [PATCH] binutils 2 39 dont error on missing makeinfo

diff --git a/bfd/Makefile.in b/bfd/Makefile.in
--- a/bfd/Makefile.in
+++ b/bfd/Makefile.in
@@ -260,7 +260,7 @@ am__v_texidevnull_ = $(am__v_texidevnull
 am__v_texidevnull_0 = > /dev/null
 am__v_texidevnull_1 = 
 am__dirstamp = $(am__leading_dot)dirstamp
-INFO_DEPS = doc/bfd.info
+INFO_DEPS = 
 am__TEXINFO_TEX_DIR = $(srcdir)
 DVIS = doc/bfd.dvi
 PDFS = doc/bfd.pdf
