--- a/include/ctf.h	2025-11-10 00:00:00.000000000 +0000
+++ b/include/ctf.h	2025-11-16 22:34:45.000000000 +0000
@@ -423,6 +423,10 @@
 #define CTF_K_CONST	12	/* ctt_type is base type.  */
 #define CTF_K_RESTRICT	13	/* ctt_type is base type.  */
 #define CTF_K_SLICE	14	/* Variant data is a ctf_slice_t.  */
+#define CTF_K_DECL_TAG	15	/* Declaration tag.  Internal use only.
+				   Not valid externally until CTF V4.  */
+#define CTF_K_TYPE_TAG	16	/* Type tag.  Internal use only.
+				   Not valid externally until CTF V4.  */
 
 #define CTF_K_MAX	63	/* Maximum possible (V2) CTF_K_* value.  */
 
