From cdbe0ad758b2024555c38802ad73603567073538 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sat, 22 Nov 2025 14:23:41 +0000
Subject: [PATCH] libctf gcc 16

diff --git a/include/ctf.h b/include/ctf.h
--- a/include/ctf.h
+++ b/include/ctf.h
@@ -424,6 +424,10 @@
 #define CTF_K_CONST	12	/* ctt_type is base type.  */
 #define CTF_K_RESTRICT	13	/* ctt_type is base type.  */
 #define CTF_K_SLICE	14	/* Variant data is a ctf_slice_t.  */
+#define CTF_K_DECL_TAG	15	/* Declaration tag.  Internal use only.
+				   Not valid externally until CTF V4.  */
+#define CTF_K_TYPE_TAG	16	/* Type tag.  Internal use only.
+				   Not valid externally until CTF V4.  */
 
 #define CTF_K_MAX	63	/* Maximum possible (V2) CTF_K_* value.  */
 
