From dec0ca839dc3ae30352c4d3a953ac4a89ec80624 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Mon, 26 Mar 2012 04:08:15 +0200
Subject: [PATCH] add /run/asound.conf as a config location

Signed-off-by: Stephan Raue <stephan@openelec.tv>
---
 src/conf/alsa.conf | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/conf/alsa.conf b/src/conf/alsa.conf
index e65bf2ca..905f07fb 100644
--- a/src/conf/alsa.conf
+++ b/src/conf/alsa.conf
@@ -12,7 +12,8 @@
 			"/usr/etc/alsa/conf.d"
 			"/etc/alsa/conf.d"
 			"/etc/asound.conf|||/usr/etc/asound.conf"
-			"~/.asoundrc"
+			"/run/asound.conf"
+			"/storage/.config/asound.conf"
 			{
 				@func concat
 				strings [
-- 
2.51.0

