From a24b6d466001ff2b3fa5679e2be0c428e5d469cd Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Mon, 30 Dec 2024 13:37:16 +0000
Subject: [PATCH] use exec_prefix/lib for libdir in glew.pc.in

---
 glew.pc.in | 2 +-
 1 files changed, 1 insertion(+), 1 deletions(-)

diff --git a/glew.pc.in b/glew.pc.in
index a8925401..cb6e4fa7 100644
--- a/glew.pc.in
+++ b/glew.pc.in
@@ -1,6 +1,6 @@
 prefix=@prefix@
 exec_prefix=${prefix}
-libdir=@libdir@
+libdir=${exec_prefix}/lib
 includedir=${prefix}/include
 
 Name: glew
