From b2881a1cb8328399302518802d3ecc8facdf30b2 Mon Sep 17 00:00:00 2001
From: heitbaum <rudi@heitbaum.com>
Date: Sun, 12 Sep 2021 11:56:35 +1000
Subject: [PATCH] add required static libaries to pkg config

diff --git a/sndfile.pc.in b/sndfile.pc.in
--- a/sndfile.pc.in
+++ b/sndfile.pc.in
@@ -8,6 +8,6 @@
 Requires:
 Requires.private: @EXTERNAL_XIPH_REQUIRE@ @EXTERNAL_MPEG_REQUIRE@
 Version: @VERSION@
-Libs: -L${libdir} -lsndfile
+Libs: -L${libdir} -lsndfile -lFLAC -lvorbis -logg -lvorbisenc -lopus
 Libs.private: @EXTERNAL_MPEG_LIBS@
 Cflags: -I${includedir}
