From 3f7fead6dd3730ab2c6f2c90b8f79fc085f5cba6 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sat, 25 Jan 2014 01:03:30 +0200
Subject: [PATCH] crosscompile badness

Index: gcc-4.4+svnr145550/gcc/incpath.cc
===================================================================
diff --git a/gcc/incpath.cc b/gcc/incpath.cc
--- a/gcc/incpath.cc
+++ b/gcc/incpath.cc
@@ -481,6 +481,26 @@ add_path (char *path, incpath_kind chain
   p->construct = 0;
   p->user_supplied_p = user_supplied_p;
 
+    /* A common error when cross compiling is including
+    host headers. This code below will try to fail fast
+    for cross compiling. Currently we consider /usr/include,
+    /opt/include and /sw/include as harmful.  */
+
+	{
+		/* printf("Adding Path: %s\n", p->name ); */
+		if( strstr(p->name, "/usr/include" ) == p->name ) {
+			fprintf(stderr, _("CROSS COMPILE Badness: /usr/include in INCLUDEPATH: %s\n"), p->name);
+			abort();
+		} else if( strstr(p->name, "/sw/include") == p->name ) {
+			fprintf(stderr, _("CROSS COMPILE Badness: /sw/include in INCLUDEPATH: %s\n"), p->name);
+			abort();
+		} else if( strstr(p->name, "/opt/include") == p->name ) {
+			fprintf(stderr, _("CROSS COMPILE Badness: /opt/include in INCLUDEPATH: %s\n"), p->name);
+			abort();
+		 }
+	}
+
+
   add_cpp_dir_path (p, chain);
 }
 
