From 719a1064b41517ab5c0ce0264b1c9098b1c0211f Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Fri, 28 Feb 2025 21:51:37 +0000
Subject: [PATCH] gettext fix

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -8,7 +8,7 @@
 AM_MAINTAINER_MODE([enable])
 
 AM_GNU_GETTEXT([external])
-AM_GNU_GETTEXT_VERSION([0.22.5])
+AM_GNU_GETTEXT_VERSION([0.24])
 
 dnl Checks for programs.
 
