From 3321938791382e56203a02aa5a2476fb538f1289 Mon Sep 17 00:00:00 2001
From: Portisch <hugo.portisch@yahoo.de>
Date: Tue, 3 Mar 2020 08:28:01 +0100
Subject: [PATCH] IsActiveDeviceType: poll only matching device type

---
 src/libcec/CECClient.cpp | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/src/libcec/CECClient.cpp b/src/libcec/CECClient.cpp
index bb8b0cd..32f14f8 100644
--- a/src/libcec/CECClient.cpp
+++ b/src/libcec/CECClient.cpp
@@ -1459,9 +1459,17 @@ bool CCECClient::IsActiveDeviceType(const cec_device_type type)
 {
   CECDEVICEVEC activeDevices;
   if (m_processor)
-    m_processor->GetDevices()->GetActive(activeDevices);
-  CCECDeviceMap::FilterType(type, activeDevices);
-  return !activeDevices.empty();
+    m_processor->GetDevices()->GetByType(type, activeDevices);
+
+  for (CECDEVICEVEC::iterator it = activeDevices.begin(); it != activeDevices.end(); it++)
+  {
+    cec_bus_device_status status = (*it)->GetStatus(false, true);
+    if (status == CEC_DEVICE_STATUS_HANDLED_BY_LIBCEC ||
+        status == CEC_DEVICE_STATUS_PRESENT)
+      return true;
+  }
+
+  return false;
 }
 
 cec_logical_address CCECClient::GetActiveSource(void)
-- 
2.17.1

