From e566a6432fd1951417bf78280cc7c741957fc419 Mon Sep 17 00:00:00 2001
From: Giovanni Cascione <ing.cascione@gmail.com>
Date: Thu, 22 May 2025 20:51:47 +0200
Subject: [PATCH] LIBRETRO: BUILD: fix armv7 GCC 15.1 internal compiler error

---
 backends/platform/libretro/Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/backends/platform/libretro/Makefile b/backends/platform/libretro/Makefile
index 0181825bfbbc..383a25cab706 100644
--- a/backends/platform/libretro/Makefile
+++ b/backends/platform/libretro/Makefile
@@ -552,6 +552,8 @@ ifneq (,$(findstring $(platform), wiiu vita))
    DEFINES += -Os
 else ifneq (,$(findstring $(platform), ios osx))
    DEFINES += -O1
+else ifneq (,$(findstring armv7, $(platform))) # fixes a GCC 15.1 internal compiler error. TODO check if it is fixed in GCC newer releases.
+   DEFINES += -O1
 else ifneq (,$(findstring $(platform), msvc genode rpi))
    DEFINES += -O2
 else
