From b1feb5dbe365179cf1e5dc210ba4f180044bae55 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Fri, 17 May 2024 14:49:46 +0000
Subject: [PATCH]  fix gcc14 build

diff --git a/tests/async_queries.c b/tests/async_queries.c
--- a/tests/async_queries.c
+++ b/tests/async_queries.c
@@ -31,7 +31,9 @@
 #include <stdio.h>
 #include <string.h>
 
-#include <event.h>
+#include <event2/event_struct.h>
+#include <event2/event_compat.h>
+#include <event2/event.h>
 
 #define SL(s) (s), sizeof(s)
 static const char *my_groups[]= { "client", NULL };
