diff -ruN a/intl/msg_hash_es.h b/intl/msg_hash_es.h
--- a/intl/msg_hash_es.h	2020-01-20 12:42:21.233274329 -0600
+++ b/intl/msg_hash_es.h	2020-01-20 12:43:39.908073357 -0600
@@ -1418,7 +1418,7 @@
     MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
     "Pausar"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
     "Reiniciar RetroArch"
@@ -2487,7 +2487,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "Reiniciar RetroArch"
diff -ruN a/intl/msg_hash_fr.h b/intl/msg_hash_fr.h
--- a/intl/msg_hash_fr.h	2020-01-20 12:42:21.433271274 -0600
+++ b/intl/msg_hash_fr.h	2020-01-20 12:43:40.120070122 -0600
@@ -1418,7 +1418,7 @@
     MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
     "Mettre en pause/reprendre"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
     "Redémarrer RetroArch"
@@ -2479,7 +2479,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "Redémarrer RetroArch"
diff -ruN a/intl/msg_hash_ja.h b/intl/msg_hash_ja.h
--- a/intl/msg_hash_ja.h	2020-01-20 12:42:21.293273412 -0600
+++ b/intl/msg_hash_ja.h	2020-01-20 12:43:39.988072138 -0600
@@ -1352,7 +1352,7 @@
    MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
    "一時停止"
 )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
    "RetroArchを終了"
@@ -2274,7 +2274,7 @@
    MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
    "MIDI"
 )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
    MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
    "再起動"
diff -ruN a/intl/msg_hash_ko.h b/intl/msg_hash_ko.h
--- a/intl/msg_hash_ko.h	2020-01-20 12:42:21.609268588 -0600
+++ b/intl/msg_hash_ko.h	2020-01-20 12:43:40.304067316 -0600
@@ -1343,7 +1343,7 @@
     MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
     "일시정지 켜기/끄기"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
     "RetroArch 재시작"
@@ -2260,7 +2260,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "RetroArch 다시 시작"
diff -ruN a/intl/msg_hash_pt_br.h b/intl/msg_hash_pt_br.h
--- a/intl/msg_hash_pt_br.h	2020-01-20 12:42:21.361272374 -0600
+++ b/intl/msg_hash_pt_br.h	2020-01-20 12:43:40.052071161 -0600
@@ -1362,7 +1362,7 @@
     MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
     "Alternar pausa"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
     "Reiniciar o RetroArch"
@@ -2419,7 +2419,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "Reiniciar o RetroArch"
diff -ruN a/intl/msg_hash_tr.h b/intl/msg_hash_tr.h
--- a/intl/msg_hash_tr.h	2020-01-20 12:42:21.505270175 -0600
+++ b/intl/msg_hash_tr.h	2020-01-20 12:43:40.180069208 -0600
@@ -1339,7 +1339,7 @@
     MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
     "Pause açma-kapama"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
     "RetroArch'ı yeniden başlat"
@@ -2241,7 +2241,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "RetroArch'ı Yeniden Başlat"
