diff -Naur a/intl/msg_hash_ar.h b/intl/msg_hash_ar.h
--- a/intl/msg_hash_ar.h	2019-01-15 12:55:14.000000000 -0600
+++ b/intl/msg_hash_ar.h	2019-01-28 22:00:41.340468631 -0600
@@ -893,7 +893,7 @@
       "Overlay next")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
       "Pause toggle")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
       "Restart RetroArch")
 #else
@@ -2672,7 +2672,7 @@
       "Show network interface(s) and associated IP addresses.")
 MSG_HASH(MENU_ENUM_SUBLABEL_SYSTEM_INFORMATION,
       "Show information specific to the device.")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_SUBLABEL_QUIT_RETROARCH,
       "Restart the program.")
 #else
@@ -3352,7 +3352,7 @@
       "Show Help")
 MSG_HASH(MENU_ENUM_SUBLABEL_MENU_SHOW_HELP,
       "Show/hide the 'Help' option.")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_LABEL_VALUE_MENU_SHOW_QUIT_RETROARCH,
       "Show Restart RetroArch")
 MSG_HASH(MENU_ENUM_SUBLABEL_MENU_SHOW_QUIT_RETROARCH,
diff -Naur a/intl/msg_hash_cht.h b/intl/msg_hash_cht.h
--- a/intl/msg_hash_cht.h	2019-01-15 12:55:14.000000000 -0600
+++ b/intl/msg_hash_cht.h	2019-01-28 22:00:41.316469044 -0600
@@ -3152,7 +3152,7 @@
       "顯示 Help")
 MSG_HASH(MENU_ENUM_SUBLABEL_MENU_SHOW_HELP,
       "顯示/隱藏 the 'Help' option.")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_LABEL_VALUE_MENU_SHOW_QUIT_RETROARCH,
       "顯示 Restart RetroArch")
 MSG_HASH(MENU_ENUM_SUBLABEL_MENU_SHOW_QUIT_RETROARCH,
diff -Naur a/intl/msg_hash_eo.h b/intl/msg_hash_eo.h
--- a/intl/msg_hash_eo.h	2019-01-15 12:55:14.000000000 -0600
+++ b/intl/msg_hash_eo.h	2019-01-28 22:00:41.344468562 -0600
@@ -750,7 +750,7 @@
       "Overlay next")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
       "Pause toggle")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
       "Restart RetroArch")
 #else
@@ -1065,7 +1065,7 @@
       "Present")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_PRIVACY_SETTINGS,
       "Privacy")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
       "Restart RetroArch")
 #else
@@ -2380,7 +2380,7 @@
       "Show network interface(s) and associated IP addresses.")
 MSG_HASH(MENU_ENUM_SUBLABEL_SYSTEM_INFORMATION,
       "Show information specific to the device.")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_SUBLABEL_QUIT_RETROARCH,
       "Restart the program.")
 #else
@@ -3040,7 +3040,7 @@
       "Show Help")
 MSG_HASH(MENU_ENUM_SUBLABEL_MENU_SHOW_HELP,
       "Show/hide the 'Help' option.")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_LABEL_VALUE_MENU_SHOW_QUIT_RETROARCH,
       "Show Restart RetroArch")
 MSG_HASH(MENU_ENUM_SUBLABEL_MENU_SHOW_QUIT_RETROARCH,
diff -Naur a/intl/msg_hash_es.h b/intl/msg_hash_es.h
--- a/intl/msg_hash_es.h	2019-01-15 12:55:14.000000000 -0600
+++ b/intl/msg_hash_es.h	2019-01-28 22:00:41.296469387 -0600
@@ -1321,7 +1321,7 @@
     MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
     "Pausar"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
     "Reiniciar RetroArch"
@@ -2080,7 +2080,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "Reiniciar RetroArch"
@@ -4957,7 +4957,7 @@
     MENU_ENUM_SUBLABEL_SYSTEM_INFORMATION,
     "Mostrar información especifica del dispositivo"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_SUBLABEL_QUIT_RETROARCH,
     "Reiniciar el programa"
@@ -6155,7 +6155,7 @@
     MENU_ENUM_SUBLABEL_MENU_SHOW_HELP,
     "Mostrar/ocultar la opción de 'Ayuda'"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_MENU_SHOW_QUIT_RETROARCH,
     "Mostrar Reiniciar RetroArch"
diff -Naur a/intl/msg_hash_fr.h b/intl/msg_hash_fr.h
--- a/intl/msg_hash_fr.h	2019-01-15 12:55:14.000000000 -0600
+++ b/intl/msg_hash_fr.h	2019-01-28 22:00:41.332468768 -0600
@@ -2359,7 +2359,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "Redémarrer RetroArch"
@@ -5786,7 +5786,7 @@
     MENU_ENUM_SUBLABEL_SYSTEM_INFORMATION,
     "Affiche les informations spécifiques à l'appareil."
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_SUBLABEL_QUIT_RETROARCH,
     "Redémarrer le programme."
@@ -7060,7 +7060,7 @@
     MENU_ENUM_SUBLABEL_MENU_SHOW_HELP,
     "Afficher/masquer l'option 'Aide'."
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_MENU_SHOW_QUIT_RETROARCH,
     "Afficher 'Redémarrer RetroArch'"
diff -Naur a/intl/msg_hash_pt_br.h b/intl/msg_hash_pt_br.h
--- a/intl/msg_hash_pt_br.h	2019-01-24 22:20:20.000000000 -0600
+++ b/intl/msg_hash_pt_br.h	2019-01-28 22:00:41.312469113 -0600
@@ -1325,7 +1325,7 @@
     MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
     "Alternar pausa"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
     "Reiniciar o RetroArch"
@@ -2084,7 +2084,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "Reiniciar o RetroArch"
@@ -5101,7 +5101,7 @@
     MENU_ENUM_SUBLABEL_SYSTEM_INFORMATION,
     "Exibe informações específicas do dispositivo."
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_SUBLABEL_QUIT_RETROARCH,
     "Rainicia o programa."
@@ -6299,7 +6299,7 @@
     MENU_ENUM_SUBLABEL_MENU_SHOW_HELP,
     "Exibir/ocultar a opção 'Ajuda'."
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_MENU_SHOW_QUIT_RETROARCH,
     "Exibir Sair do RetroArch"
diff -Naur a/intl/msg_hash_us.h b/intl/msg_hash_us.h
--- a/intl/msg_hash_us.h	2019-01-28 17:00:44.000000000 -0600
+++ b/intl/msg_hash_us.h	2019-01-28 22:00:41.384467872 -0600
@@ -1329,7 +1329,7 @@
     MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
     "Pause toggle"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
     "Restart RetroArch"
@@ -2104,7 +2104,7 @@
     MENU_ENUM_LABEL_VALUE_MIDI_SETTINGS,
     "MIDI"
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_QUIT_RETROARCH,
     "Restart RetroArch"
@@ -5141,7 +5141,7 @@
     MENU_ENUM_SUBLABEL_SYSTEM_INFORMATION,
     "Show information specific to the device."
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_SUBLABEL_QUIT_RETROARCH,
     "Restart the program."
@@ -6339,7 +6339,7 @@
     MENU_ENUM_SUBLABEL_MENU_SHOW_HELP,
     "Show/hide the 'Help' option."
     )
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(
     MENU_ENUM_LABEL_VALUE_MENU_SHOW_QUIT_RETROARCH,
     "Show Restart RetroArch"
diff -Naur a/intl/msg_hash_vn.h b/intl/msg_hash_vn.h
--- a/intl/msg_hash_vn.h	2019-01-15 12:55:14.000000000 -0600
+++ b/intl/msg_hash_vn.h	2019-01-28 22:00:41.368468148 -0600
@@ -833,7 +833,7 @@
       "Overlay next")
 MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_META_PAUSE_TOGGLE,
       "Pause toggle")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_LABEL_VALUE_INPUT_META_QUIT_KEY,
       "Restart RetroArch")
 #else
@@ -2542,7 +2542,7 @@
       "Show network interface(s) and associated IP addresses.")
 MSG_HASH(MENU_ENUM_SUBLABEL_SYSTEM_INFORMATION,
       "Show information specific to the device.")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_SUBLABEL_QUIT_RETROARCH,
       "Restart the program.")
 #else
@@ -3194,7 +3194,7 @@
       "Show Help")
 MSG_HASH(MENU_ENUM_SUBLABEL_MENU_SHOW_HELP,
       "Show/hide the 'Help' option.")
-#ifdef HAVE_LAKKA
+#ifndef HAVE_LAKKA
 MSG_HASH(MENU_ENUM_LABEL_VALUE_MENU_SHOW_QUIT_RETROARCH,
       "Show Restart RetroArch")
 MSG_HASH(MENU_ENUM_SUBLABEL_MENU_SHOW_QUIT_RETROARCH,
