--- a/cheevos/cheevos.c
+++ b/cheevos/cheevos.c
@@ -304,6 +304,12 @@
       cheevo->active &= ~RCHEEVOS_ACTIVE_HARDCORE;
 
    cheevo->unlock_time = cpu_features_get_time_usec();
+
+    char pixelcade[256];
+    snprintf(pixelcade, sizeof(pixelcade), "/storage/.emulationstation/scripts/achievements/pixelcade.sh %u \"%s\" \"%s\" > /dev/null 2>/dev/null",
+    cheevo->id, cheevo->title, cheevo->description);
+    //CHEEVOS_LOG(pixelcade);
+    system(pixelcade);
 
    /* Show the on screen message. */
 #if defined(HAVE_GFX_WIDGETS)
