From 227fb82c2cf4e0f47293a8ff5451e5d71e3ba224 Mon Sep 17 00:00:00 2001
From: portisch <hugo.portisch@yahoo.de>
Date: Wed, 22 May 2024 11:02:51 +0200
Subject: [PATCH 05/34] DVDVideoCodecAmlogic: use correct decoder for MVC

---
 .../VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp     | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
index 87788d6de9..5be949f018 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
@@ -173,7 +173,10 @@ bool CDVDVideoCodecAmlogic::Open(CDVDStreamInfo &hints, CDVDCodecOptions &option
         m_h264_sequence->ratio  = m_hints.aspect;
       }
 
-      m_pFormatName = "am-h264";
+      if (m_hints.codec_tag == MKTAG('M', 'V', 'C', '1'))
+        m_pFormatName = "am-h264mvc";
+      else
+        m_pFormatName = "am-h264";
       // convert h264-avcC to h264-annex-b as h264-avcC
       // under streamers can have issues when seeking.
       if (m_hints.extradata && m_hints.extradata.GetData()[0] == 1)
-- 
2.34.1

