From 62cc537e4bd1f47a6cbeb91c1bead786e88ca6ab Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Sat, 3 Aug 2019 20:58:07 +0100
Subject: [PATCH 09/34] RenderFlags: define FP mode

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/cores/VideoPlayer/VideoRenderers/RenderFlags.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/RenderFlags.h b/xbmc/cores/VideoPlayer/VideoRenderers/RenderFlags.h
index 54f0524848..ae9ec99dc1 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/RenderFlags.h
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/RenderFlags.h
@@ -57,6 +57,7 @@
 #define CONF_FLAGS_STEREO_MODE_MASK(a) ((a) & 0x007000)
 #define CONF_FLAGS_STEREO_MODE_SBS     0x001000
 #define CONF_FLAGS_STEREO_MODE_TAB     0x002000
+#define CONF_FLAGS_STEREO_MODE_FP      0x004000
 
 #define CONF_FLAGS_STEREO_CADENCE(a) ((a) & 0x008000)
 #define CONF_FLAGS_STEREO_CADANCE_LEFT_RIGHT 0x000000
-- 
2.34.1

