From e489ceddeb34ef7c9e9e61289f4008f43a67acfc Mon Sep 17 00:00:00 2001
From: portisch <hugo.portisch@yahoo.de>
Date: Wed, 29 May 2024 14:12:00 +0200
Subject: [PATCH 21/34] AMLUtils: add probe support for frame packaging modes

---
 xbmc/utils/AMLUtils.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index a79ca48e10..7601ca4b19 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -468,6 +468,15 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
   res->strMode       = StringUtils::Format("{:d}x{:d} @ {:.2f}{} - Full Screen", res->iScreenWidth, res->iScreenHeight, res->fRefreshRate,
     res->dwFlags & D3DPRESENTFLAG_INTERLACED ? "i" : "");
 
+  if (fromMode.find("FramePacking") != std::string::npos)
+    res->dwFlags |= D3DPRESENTFLAG_MODE3DFP;
+
+  if (fromMode.find("TopBottom") != std::string::npos)
+    res->dwFlags |= D3DPRESENTFLAG_MODE3DTB;
+
+  if (fromMode.find("SidebySide") != std::string::npos)
+    res->dwFlags |= D3DPRESENTFLAG_MODE3DSBS;
+
   return res->iWidth > 0 && res->iHeight> 0;
 }
 
-- 
2.45.2

