From 6d57bc179224d814f1c7b336c6dbc44e23fc1d45 Mon Sep 17 00:00:00 2001
From: portisch <hugo.portisch@yahoo.de>
Date: Wed, 29 May 2024 14:12:00 +0200
Subject: [PATCH 21/34] AMLUtils: add probe support for frame packaging modes

---
 xbmc/utils/AMLUtils.cpp | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index c29767d075..51e9404fc4 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -466,6 +466,15 @@ bool aml_mode_to_resolution(const char *mode, RESOLUTION_INFO *res)
   res->strMode       = StringUtils::Format("{:d}x{:d} @ {:.2f}{} - Full Screen", res->iScreenWidth, res->iScreenHeight, res->fRefreshRate,
     res->dwFlags & D3DPRESENTFLAG_INTERLACED ? "i" : "");
 
+  if (fromMode.find("FramePacking") != std::string::npos)
+    res->dwFlags |= D3DPRESENTFLAG_MODE3DFP;
+
+  if (fromMode.find("TopBottom") != std::string::npos)
+    res->dwFlags |= D3DPRESENTFLAG_MODE3DTB;
+
+  if (fromMode.find("SidebySide") != std::string::npos)
+    res->dwFlags |= D3DPRESENTFLAG_MODE3DSBS;
+
   return res->iWidth > 0 && res->iHeight> 0;
 }
 
-- 
2.34.1

