From 1305a28b37085229d804440717fc0f7ca16a3949 Mon Sep 17 00:00:00 2001
From: portisch <hugo.portisch@yahoo.de>
Date: Thu, 11 Jul 2024 10:25:47 +0200
Subject: [PATCH 34/34] StereoscopicsManager: cover missing playback ended

---
 xbmc/guilib/StereoscopicsManager.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/xbmc/guilib/StereoscopicsManager.cpp b/xbmc/guilib/StereoscopicsManager.cpp
index d14d68251b..9d9e43f1ed 100644
--- a/xbmc/guilib/StereoscopicsManager.cpp
+++ b/xbmc/guilib/StereoscopicsManager.cpp
@@ -402,6 +402,7 @@ bool CStereoscopicsManager::OnMessage(CGUIMessage &message)
 {
   switch (message.GetMessage())
   {
+  case GUI_MSG_PLAYBACK_ENDED:
   case GUI_MSG_PLAYBACK_STOPPED:
   case GUI_MSG_PLAYLISTPLAYER_STOPPED:
     OnPlaybackStopped();
-- 
2.45.2

