From c959a52b760e234a3a1b544a07d3c51dbd8b9047 Mon Sep 17 00:00:00 2001
From: roidy <roidy@users.noreply.github.com>
Date: Sat, 9 May 2020 17:53:27 +0100
Subject: [PATCH] Fix for <fanartres> not being used in advancedsettings.xml

---
 xbmc/pictures/Picture.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/xbmc/pictures/Picture.cpp b/xbmc/pictures/Picture.cpp
index 3ed18e8ad4..9816ae6a1d 100644
--- a/xbmc/pictures/Picture.cpp
+++ b/xbmc/pictures/Picture.cpp
@@ -214,12 +214,13 @@ bool CPicture::CacheTexture(uint8_t *pixels, uint32_t width, uint32_t height, ui
   uint32_t max_height = advancedSettings->m_imageRes;
   if (advancedSettings->m_fanartRes > advancedSettings->m_imageRes)
   { // 16x9 images larger than the fanart res use that rather than the image res
-    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f) <= 0.01f &&
-        height >= advancedSettings->m_fanartRes)
+    if (fabsf(static_cast<float>(width) / static_cast<float>(height) / (16.0f / 9.0f) - 1.0f)
+        <= 0.01f)
     {
-      max_height = advancedSettings->m_fanartRes;
+      max_height = advancedSettings->m_fanartRes; // use height defined in fanartRes
     }
   }
+
   uint32_t max_width = max_height * 16/9;
 
   dest_height = std::min(dest_height, max_height);
-- 
2.25.1

