--- a/xbmc/filesystem/CurlFile.cpp
+++ b/xbmc/filesystem/CurlFile.cpp
@@ -1000,6 +1000,11 @@
 
   // setup common curl options
   SetCommonOptions(m_state, m_failOnError && !CServiceBroker::GetSettingsComponent()->GetAdvancedSettings()->CanLogComponent(LOGCURL));
+
+  if(strcmp(url2.GetHostName().c_str(), "libreelec.tv")
+    || strcmp(url2.GetHostName().c_str(), "kodi.tv"))
+    g_curlInterface.easy_setopt(m_state->m_easyHandle, CURLOPT_USERAGENT, "Kodi/18.3 (X11; Linux aarch64) LibreELEC/9.0 App_Bitness/32 Version/18.3-Git:18.3-Leia");
+
   SetRequestHeaders(m_state);
   m_state->m_sendRange = m_seekable;
   m_state->m_bRetry = m_allowRetry;
--- a/xbmc/utils/SystemInfo.cpp
+++ b/xbmc/utils/SystemInfo.cpp
@@ -1194,7 +1194,7 @@
   StringUtils::Replace(fullVer, ' ', '-');
   result += " Version/" + fullVer;

-  return result;
+  return "Kodi/18.3 (X11; Linux aarch64) LibreELEC/9.0 App_Bitness/32 Version/18.3-Git:18.3-Leia";
 }

 std::string CSysInfo::GetDeviceName()
