--- a/xbmc/windowing/amlogic/AMLDisplay.cpp
+++ b/xbmc/windowing/amlogic/AMLDisplay.cpp
@@ -167,9 +167,6 @@ CAMLDRMUtils::CAMLDRMUtils()
   }
 
   aml_init_drmDevice();
-
-  if (aml_get_drmDevice_connected())
-    aml_init_drmDevice_display();
 }
 
 CAMLDRMUtils::~CAMLDRMUtils()
@@ -190,23 +187,43 @@ CAMLDRMUtils::~CAMLDRMUtils()
 
 void CAMLDRMUtils::CleanAndClose()
 {
   if (m_resources)
+  {
     drmModeFreeResources(m_resources);
+    m_resources = nullptr;
+  }
 
   if (m_connector)
+  {
     drmModeFreeConnector(m_connector);
+    m_connector = nullptr;
+  }
 
   if (m_encoder)
+  {
     drmModeFreeEncoder(m_encoder);
+    m_encoder = nullptr;
+  }
 
   if (m_crtc)
+  {
     drmModeFreeCrtc(m_crtc);
+    m_crtc = nullptr;
+  }
 
   if (m_orig_crtc)
+  {
     free(m_orig_crtc);
+    m_orig_crtc = nullptr;
+  }
 
   if (m_plane)
+  {
     drmModeFreePlane(m_plane);
+    m_plane = nullptr;
+  }
+
+  m_connection = DRM_MODE_DISCONNECTED;
 }
 
 void CAMLDRMUtils::aml_init_drmDevice()
@@ -279,5 +298,8 @@ void CAMLDRMUtils::aml_init_drmDevice()
     throw std::runtime_error("failed to get primary plane of drmDevice");
   }
+
+  if (aml_get_drmDevice_connected())
+    aml_init_drmDevice_display();
 }
 
 void CAMLDRMUtils::aml_init_drmDevice_display()
