From 60d8f2245761d64546eeb2eb3f44fe89984cf08d Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Sun, 24 Mar 2013 15:06:00 +0200
Subject: [PATCH] silence missing identity error

diff --git a/sshconnect2.c b/sshconnect2.c
index d1555ee..25cc9e5 100644
--- a/sshconnect2.c
+++ b/sshconnect2.c
@@ -1535,9 +1535,7 @@ load_identity_file(Identity *id)
 	struct stat st;
 
 	if (stat(id->filename, &st) == -1) {
-		do_log2(id->userprovided ?
-		    SYSLOG_LEVEL_INFO : SYSLOG_LEVEL_DEBUG3,
-		    "no such identity: %s: %s", id->filename, strerror(errno));
+		debug3("no such identity: %s", id->filename);
 		return NULL;
 	}
 	snprintf(prompt, sizeof prompt,
-- 
2.53.0

