From c05ee9208db5e25d550ec6dd4b5ff54a199959fd Mon Sep 17 00:00:00 2001
From: MilhouseVH <milhouseVH.github@nmacleod.com>
Date: Mon, 16 Dec 2019 20:34:22 +0000
Subject: [PATCH] openssh: source /etc/environment

---
 session.c | 2 --
 1 file changed, 2 deletions(-)

--- a/session.c
+++ b/session.c
@@ -1036,7 +1036,6 @@ do_setup_env(struct ssh *ssh, Session *s
 			child_set_env(&env, &envsize, "KRB5CCNAME", cp);
 	}
 
-#ifdef _AIX
 	{
 		char *cp;
 
@@ -1045,7 +1044,6 @@ do_setup_env(struct ssh *ssh, Session *s
 		read_environment_file(&env, &envsize, "/etc/environment",
 		    options.permit_user_env_allowlist);
 	}
-#endif
 #ifdef KRB5
 	if (s->authctxt->krb5_ccname)
 		child_set_env(&env, &envsize, "KRB5CCNAME",
