From 8b36cc8aba405a06f6741622e65ac4e87fe6ea99 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Wed, 01 Feb 2017 13:39:10 -0800
Subject: [PATCH] disable building ffmvforce

diff --git a/utils/Makefile b/utils/Makefile
--- a/utils/Makefile
+++ b/utils/Makefile
@@ -37,7 +37,7 @@
 endif
 
 ifndef DISABLE_FORCEFEEDBACK
-PROGRAMS	+= fftest ffmvforce ffset ffcfstress
+PROGRAMS	+= fftest ffset ffcfstress
 endif
 
 PREFIX          ?= /usr/local
