From 2e94f15c5206f15d0770e91144c481784818e7fb Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Fri, 16 Oct 2015 11:42:39 -0700
Subject: [PATCH] check uid

diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -1462,10 +1462,6 @@ static int check_ours(const char *p) {
     if (stat(p, &st) < 0)
         return -errno;
 
-#ifdef HAVE_GETUID
-    if (st.st_uid != getuid() && st.st_uid != 0)
-        return -EACCES;
-#endif
 
     return 0;
 }
