--- a/es-core/src/resources/Font.cpp
+++ b/es-core/src/resources/Font.cpp
@@ -258,9 +258,9 @@
 	// Linux
 
 	const char* paths[] = { 
-		"/usr/share/fonts/truetype/dejavu/DejaVuSans.ttf",
-		"/usr/share/fonts/truetype/freefont/FreeMono.ttf",
-		"/usr/share/fonts/truetype/droid/DroidSansFallbackFull.ttf" // japanese, chinese, present on Debian
+		"/storage/.config/emulationstation/resources/fonts/truetype/dejavu/DejaVuSans.ttf",
+		"/storage/.config/emulationstation/resources/fonts/truetype/freefont/FreeMono.ttf",
+		"/storage/.config/emulationstation/resources/fonts/truetype/droid/DroidSansFallbackFull.ttf" // japanese, chinese, present on Debian
 	};
 
 	std::vector<std::string> fontPaths;
