--- a/Makefile
+++ b/Makefile
@@ -245,7 +245,7 @@
 
 # (armv8 a35, hard point, neon based) ###
 # Playstation Classic
-else ifeq ($(platform), classic_armv8_a35)
+else ifeq ($(platform), emuelec)
 	TARGET := $(TARGET_NAME)_libretro.so
 	fpic := -fPIC
     SHARED := -shared -Wl,-version-script=$(LIBRETRO_DIR)/link.T -Wl,-no-undefined
@@ -256,9 +256,9 @@
 	-falign-functions=1 -falign-jumps=1 -falign-loops=1 \
 	-fno-unwind-tables -fno-asynchronous-unwind-tables -fno-unroll-loops \
 	-fmerge-all-constants -fno-math-errno \
-	-marm -mtune=cortex-a35 -mfpu=neon-fp-armv8 -mfloat-abi=hard
+	-mtune=cortex-a72 #-mfpu=neon-fp-armv8 -mfloat-abi=hard
 	HAVE_NEON = 1
-	ARCH = arm
+	ARCH = aarch64
 	ifeq ($(shell echo `$(CC) -dumpversion` "< 4.9" | bc -l), 1)
 	CFLAGS += -march=armv8-a
 	else
