--- a/tasks/task_content.c	2019-01-15 12:55:14.000000000 -0600
+++ b/tasks/task_content.c	2019-01-28 22:00:41.340468631 -0600
@@ -261,7 +261,9 @@
 
    content_info->data = content_data;
    content_info->size = length;
-
+   
+   system("/usr/bin/fbfix"); // emuelec
+   
    return true;
 }
 
@@ -786,6 +788,7 @@
    {
       RARCH_LOG("[CORE]: Using content: %s.\n", args->content_path);
       argv[(*argc)++] = strdup(args->content_path);
+    system("/usr/bin/fbfix"); // emuelec
    }
 #ifdef HAVE_MENU
    else
@@ -1123,7 +1126,7 @@
 #else
    frontend_driver_set_fork(FRONTEND_FORK_CORE_WITH_ARGS);
 #endif
-
+	system("/usr/bin/fbfix"); // emuelec
    return true;
 }
 #endif

--- a/retroarch.c	2019-01-15 12:55:14.000000000 -0600
+++ b/retroarch.c	2019-01-28 22:00:41.340468631 -0600 
@@ -12264,6 +12264,7 @@
       input_remapping_set_defaults();
    }
 #endif
+system("/usr/bin/fbfix"); // emuelec
 }
 
 #ifdef HAVE_CHEATS
@@ -15000,6 +15001,8 @@
                RARCH_LOG("[Input]: %s => %s\n",
                      "Game Focus",
                      p_rarch->game_focus_state.enabled ? "ON" : "OFF");
+                    
+                    system("/usr/bin/fbfix"); // emuelec   
             }
          }
          break;
