--- a/tasks/task_content.c	2019-01-15 12:55:14.000000000 -0600
+++ b/tasks/task_content.c	2019-01-28 22:00:41.340468631 -0600
@@ -789,7 +789,9 @@
 
    *data      = content_data;
    *data_size = (size_t)content_size;
-
+    
+    system("/usr/bin/fbfix"); // emuelec
+    
    return true;
 }
 
@@ -1368,6 +1370,7 @@
    {
       RARCH_LOG("[CORE]: Using content: %s.\n", args->content_path);
       argv[(*argc)++] = strdup(args->content_path);
+    system("/usr/bin/fbfix"); // emuelec
    }
 #ifdef HAVE_MENU
    else
@@ -1705,7 +1708,7 @@
 #else
    frontend_driver_set_fork(FRONTEND_FORK_CORE_WITH_ARGS);
 #endif
-
+	system("/usr/bin/fbfix"); // emuelec
    return true;
 }
 #endif

--- a/retroarch.c	2019-01-15 12:55:14.000000000 -0600
+++ b/retroarch.c	2019-01-28 22:00:41.340468631 -0600 
@@ -8442,6 +8442,8 @@
                RARCH_LOG("[Input]: %s => %s\n",
                      "Game Focus",
                      input_st->game_focus_state.enabled ? "ON" : "OFF");
+                     
+                     system("/usr/bin/fbfix"); // emuelec
             }
          }
          break;
@@ -13207,6 +13209,7 @@
          ui_companion_qt.toggle(p_rarch->ui_companion_qt_data, force);
    }
 #endif
+ system("/usr/bin/fbfix"); // emuelec
 }
 
 static void ui_companion_driver_init_first(
