From 618ccd9a332b0787f96b7fdcab120d89f7b4f281 Mon Sep 17 00:00:00 2001
From: heitbaum <rudi@heitbaum.com>
Date: Sun, 10 Jan 2021 04:07:19 +0000
Subject: [PATCH] disable docs tests

diff --git a/configure.ac b/configure.ac
index e93e1a7..3a5de17 100644
--- a/configure.ac
+++ b/configure.ac
@@ -762,9 +762,7 @@ AM_CONDITIONAL([MACOSX], [test x$macosx = xtrue])
 AC_CONFIG_FILES([
    Makefile
    scripts/Makefile
-   doc/Makefile
-   doc/de/Makefile doc/es/Makefile doc/fr/Makefile doc/nl/Makefile doc/pt/Makefile doc/ro/Makefile doc/sv/Makefile
-   test/Makefile test/defs])
+   ])
 AC_OUTPUT
 
 dnl Local variables:
diff --git a/Makefile.am b/Makefile.am
index ff71a8d..f8c378d 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,6 +1,6 @@
 AUTOMAKE_OPTIONS=foreign
 ACLOCAL_AMFLAGS = -I build-aux
-SUBDIRS=doc scripts test
+SUBDIRS=scripts
 
 noinst_LTLIBRARIES = libcommunicate.la libmacosx.la libfakeroot_time64.la
 libcommunicate_la_SOURCES = communicate.c
-- 
2.53.0

