--- a/server/streamreader/airplay_stream.cpp	2024-04-14 10:33:22.000000000 +0000
+++ b/server/streamreader/airplay_stream.cpp	2024-04-14 14:42:42.429301110 +0000
@@ -56,7 +56,7 @@
     string devicename = uri_.getQuery("devicename", "Snapcast");
     string password = uri_.getQuery("password", "");
 
-    params_wo_port_ = "\"--name=" + devicename + "\" --output=stdout --get-coverart";
+    params_wo_port_ = "\"--name=" + devicename + "\" --output=stdout --get-coverart --mdns=avahi";
     if (!password.empty())
         params_wo_port_ += " --password \"" + password + "\"";
     if (!params_.empty())
