From c1a993a4161e5ce22203383f32295ca1253a9f15 Mon Sep 17 00:00:00 2001
From: 7Ji <pugokushin@gmail.com>
Date: Mon, 27 Jun 2022 15:30:26 +0800
Subject: [PATCH] dont define log2 and log2l

---
 sources/src/include/sysdeps.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/sources/src/include/sysdeps.h b/sources/src/include/sysdeps.h
index 9220c54..4782c0f 100644
--- a/sources/src/include/sysdeps.h
+++ b/sources/src/include/sysdeps.h
@@ -747,11 +747,4 @@ typedef uint8_t uint8;
 #define _daylight 0
 #endif
 
-#ifndef log2
-#define log2 logb
-#endif
-#ifndef log2l
-#define log2l logbl
-#endif
-
 #endif /* UAE_SYSDEPS_H */
-- 
2.36.1

