--- a/desktop_version/src/FileSystemUtils.cpp
+++ b/desktop_version/src/FileSystemUtils.cpp
@@ -109,7 +109,7 @@
 	}
 	else
 	{
-		strcpy(output, PHYSFS_getBaseDir());
+		strcpy(output, "/storage/roms/ports/VVVVVV/");
 		strcat(output, "data.zip");
 	}
 	if (!PHYSFS_mount(output, NULL, 1))
@@ -130,7 +130,7 @@
 		return 0;
 	}
 
-	strcpy(output, PHYSFS_getBaseDir());
+	strcpy(output, "/storage/.config/SDL-GameControllerDB/");
 	strcpy(output, "gamecontrollerdb.txt");
 	if (SDL_GameControllerAddMappingsFromFile(output) < 0)
 	{
