From 64335282e4ae99b1ac5409442e9670c2df3421ae Mon Sep 17 00:00:00 2001
From: Portisch <hugo.portisch@yahoo.de>
Date: Tue, 19 Apr 2022 15:44:54 +0200
Subject: [PATCH] TEMP: force 'm_bStandalone' to 'true' because of issue #21284

---
 xbmc/Application.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/Application.h b/xbmc/Application.h
index 1ddf3ca954..c54008dfb8 100644
--- a/xbmc/Application.h
+++ b/xbmc/Application.h
@@ -402,7 +402,7 @@ protected:
   bool m_skipGuiRender = false;
 
   std::string m_logTarget;
-  bool m_bStandalone = false;
+  bool m_bStandalone = true;
   bool m_bTestMode = false;
   bool m_bSystemScreenSaverEnable = false;
 
-- 
2.35.1

