From 932bb3d666b6ae281dbdf265947bbc275f666dd8 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Thu, 29 Aug 2024 09:49:06 +0000
Subject: [PATCH] rename pty.h to screen-pty.h

diff --git a/display.c b/display.c
index a3fc962..a52ad1e 100644
--- a/display.c
+++ b/display.c
@@ -47,7 +47,7 @@
 #include "mark.h"
 #include "misc.h"
 #include "process.h"
-#include "pty.h"
+#include "screen-pty.h"
 #include "resize.h"
 #include "termcap.h"
 #include "tty.h"
diff --git a/Makefile.in b/Makefile.in
index b64a288..5312d97 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -67,7 +67,7 @@ screen: $(OFILES)
 	$(CC) $(LDFLAGS) -o $@ $(OFILES) $(LIBS)
 
 .c.o:
-	$(CC) -c $(CPPFLAGS) $(CFLAGS) $< -o $@
+	$(CC) -c -I$(srcdir) $(CPPFLAGS) $(CFLAGS) $< -o $@
 
 check: $(TESTBIN) $(TESTSTANDALONE)
 	for f in $(TESTBIN) $(TESTSTANDALONE); do \
@@ -205,12 +205,12 @@ search.o: search.c config.h screen.h os.h ansi.h sched.h acls.h comm.h \
  logfile.h mark.h input.h
 tty.o: tty.c config.h screen.h os.h ansi.h sched.h acls.h comm.h layer.h \
  term.h image.h canvas.h display.h layout.h viewport.h window.h logfile.h \
- fileio.h misc.h pty.h telnet.h tty.h
+ fileio.h misc.h screen-pty.h telnet.h tty.h
 term.o: term.c term.h
 window.o: window.c config.h screen.h os.h ansi.h sched.h acls.h comm.h \
  layer.h term.h image.h canvas.h display.h layout.h viewport.h window.h \
  logfile.h winmsg.h winmsgbuf.h winmsgcond.h backtick.h fileio.h help.h \
- input.h mark.h misc.h process.h pty.h resize.h telnet.h termcap.h tty.h \
+ input.h mark.h misc.h process.h screen-pty.h resize.h telnet.h termcap.h tty.h \
  utmp.h
 utmp.o: utmp.c config.h screen.h os.h ansi.h sched.h acls.h comm.h \
  layer.h term.h image.h canvas.h display.h layout.h viewport.h window.h \
@@ -237,7 +237,7 @@ process.o: process.c config.h screen.h os.h ansi.h sched.h acls.h comm.h \
 display.o: display.c config.h screen.h os.h ansi.h sched.h acls.h comm.h \
  layer.h term.h image.h canvas.h display.h layout.h viewport.h window.h \
  logfile.h winmsg.h winmsgbuf.h winmsgcond.h backtick.h encoding.h mark.h \
- misc.h process.h pty.h resize.h termcap.h tty.h
+ misc.h process.h screen-pty.h resize.h termcap.h tty.h
 comm.o: comm.c config.h os.h screen.h ansi.h sched.h acls.h comm.h \
  layer.h term.h image.h canvas.h display.h layout.h viewport.h window.h \
  logfile.h
diff --git a/pty.c b/pty.c
index 5139403..3ccb03d 100644
--- a/pty.c
+++ b/pty.c
@@ -28,7 +28,7 @@
 
 #include "config.h"
 
-#include "pty.h"
+#include "screen-pty.h"
 
 #include <sys/ioctl.h>
 
diff --git a/pty.h b/pty.h
index f72d6ea..0000000 100644
--- a/pty.h
+++ b/pty.h
@@ -1,11 +0,0 @@
-#ifndef SCREEN_PTY_H
-#define SCREEN_PTY_H
-
-int   OpenPTY (char **);
-int  ClosePTY (int);
-
-/* global variables */
-
-extern int pty_preopen;
-
-#endif /* SCREEN_PTY_H */
diff --git a/screen-pty.h b/screen-pty.h
new file mode 100644
index 0000000..f72d6ea
--- /dev/null
+++ b/screen-pty.h
@@ -0,0 +1,11 @@
+#ifndef SCREEN_PTY_H
+#define SCREEN_PTY_H
+
+int   OpenPTY (char **);
+int  ClosePTY (int);
+
+/* global variables */
+
+extern int pty_preopen;
+
+#endif /* SCREEN_PTY_H */
diff --git a/tty.c b/tty.c
index 3e0f922..ecebe84 100644
--- a/tty.c
+++ b/tty.c
@@ -49,7 +49,7 @@
 #include "screen.h"
 #include "fileio.h"
 #include "misc.h"
-#include "pty.h"
+#include "screen-pty.h"
 #include "telnet.h"
 #include "tty.h"
 
diff --git a/window.c b/window.c
index 68fc006..e97b705 100644
--- a/window.c
+++ b/window.c
@@ -48,7 +48,7 @@
 #include "mark.h"
 #include "misc.h"
 #include "process.h"
-#include "pty.h"
+#include "screen-pty.h"
 #include "resize.h"
 #include "telnet.h"
 #include "termcap.h"
-- 
2.53.0

