From 10684eea54a2c3b0b3738dfd8d70014f54e8e282 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sat, 29 Mar 2025 02:23:34 +1100
Subject: [PATCH] Fix build with libxml2-2.14.0

---
 src/xml_pyx.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/xml_pyx.c b/src/xml_pyx.c
index ab295f12..e5ec64f1 100644
--- a/src/xml_pyx.c
+++ b/src/xml_pyx.c
@@ -21,6 +21,12 @@
 
 #include "xmlstar.h"
 
+#if __GNUC__ * 100 + __GNUC_MINOR__ >= 207
+  #define ATTRIBUTE_UNUSED __attribute__((unused))
+#else
+  #define ATTRIBUTE_UNUSED
+#endif
+
 /**
  *  Output newline and tab characters as escapes
  *  Required both for attribute values and character data (#PCDATA)
