--- a/src/engine/CGameLauncher.cpp
+++ b/src/engine/CGameLauncher.cpp
@@ -217,9 +217,7 @@ bool CGameLauncher::setupMenu()
 
         ss << "No games detected!\n";
         ss << "You can copy some into:\n";
-        ss << "\"" << searchPathGames << "\",\n";
-        ss << "or download using \"+ More\" button\n";
-        ss << "on the next page.\n";
+        ss << "\"/storage/roms/ports/cgenius/games\",\n";
 
         std::string msg(ss.str());
 
