--- a/drivers/media/v4l2-core/v4l2-fwnode.c	2019-04-02 00:31:42.224001000 +0200
+++ b/drivers/media/v4l2-core/v4l2-fwnode.c	2019-04-02 00:31:42.256001000 +0200
@@ -553,6 +553,20 @@
 }
 EXPORT_SYMBOL_GPL(v4l2_fwnode_endpoint_alloc_parse);
 
+bool of_node_name_eq(const struct device_node *np, const char *name)
+{
+	const char *node_name;
+	size_t len;
+
+	if (!np)
+		return false;
+
+	node_name = kbasename(np->full_name);
+	len = strchrnul(node_name, '@') - node_name;
+
+	return (strlen(name) == len) && (strncmp(node_name, name, len) == 0);
+}
+
 int v4l2_fwnode_parse_link(struct fwnode_handle *__fwnode,
 			   struct v4l2_fwnode_link *link)
 {
