From 4d6eb422bc4905bbc370791ba3ae659f97c59399 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@siol.net>
Date: Thu, 23 Aug 2018 07:08:13 +0200
Subject: [PATCH] fix compilation issue

diff --git a/src/mod_path.c b/src/mod_path.c
--- a/src/mod_path.c
+++ b/src/mod_path.c
@@ -1,6 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/sysmacros.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <unistd.h>
