From 542342c24f1514c9fca16c064b8a4faada169b30 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Thu, 06 Jul 2023 19:23:52 -0700
Subject: [PATCH] add enable gl switch

diff --color -Naur a/configure.ac b/configure.ac
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -37,7 +37,8 @@
   enable_sdl=yes
 ], [
   dnl Running in a normal OS (not emscripten)
-  AX_CHECK_GL
+  AC_ARG_ENABLE([gl], AS_HELP_STRING([--enable-gl], [OpenGL support]), [], [enable_gl=no])
+  AS_IF([test "x$enable_gl" = "xyes"], [AX_CHECK_GL])
 
   # check OS if mac or linux
   AC_CANONICAL_HOST
