--- a/src/host/buildvm.c	2019-11-19 18:33:03.914237086 +0100
+++ b/src/host/buildvm.c	2019-11-19 18:33:08.837645902 +0100
@@ -434,12 +434,6 @@
   BuildCtx *ctx = &ctx_;
   int status, binmode;
 
-  if (sizeof(void *) != 4*LJ_32+8*LJ_64) {
-    fprintf(stderr,"Error: pointer size mismatch in cross-build.\n");
-    fprintf(stderr,"Try: make HOST_CC=\"gcc -m32\" CROSS=...\n\n");
-    return 1;
-  }
-
   UNUSED(argc);
   parseargs(ctx, argv);
 
