From 7c847c106d95d3515a075e8c63c36d789096d911 Mon Sep 17 00:00:00 2001
From: Garrett Brown <themagnificentmrb@gmail.com>
Date: Mon, 10 Apr 2023 16:12:06 +0200
Subject: [PATCH] do not use static libgcc libstdc++

diff --git a/src/os/libretro/Makefile b/src/os/libretro/Makefile
index 85c8a02f3f..ece7e8ccd7 100644
--- a/src/os/libretro/Makefile
+++ b/src/os/libretro/Makefile
@@ -58,7 +58,6 @@ ifneq ($(GIT_VERSION)," unknown")
    CXXFLAGS += -DGIT_VERSION=\"$(GIT_VERSION)\"
 endif
 
-GET_STATIC_ARG=$(if $(filter $(shell $(CXX) -print-file-name=$1.a),$1.a),,-static-$1)
 # Unix
 ifneq (,$(findstring unix,$(platform)))
    GPP_MAJOR := $(shell $(CXX) -dumpversion | cut -d '.' -f 1)
@@ -66,7 +65,7 @@ ifneq (,$(findstring unix,$(platform)))
       $(error Unable to determine $(CXX) version)
    endif
    CXXFLAGS += $(LTO)
-   LDFLAGS += $(LTO) $(PTHREAD_FLAGS) $(call GET_STATIC_ARG,libgcc) $(call GET_STATIC_ARG,libstdc++)
+   LDFLAGS += $(LTO) $(PTHREAD_FLAGS)
    TARGET := $(TARGET_NAME)_libretro.so
    fpic := -fPIC
    ifneq ($(findstring SunOS,$(shell uname -a)),)
-- 
2.53.0

