From f3bd0106abd9f4fbf8ef0e5afbe0dc1e90c065d5 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Sun, 27 Nov 2022 10:21:30 +0000
Subject: [PATCH] do not use git version

diff --git a/meson.build b/meson.build
index ad26a5a..5908ef9 100644
--- a/meson.build
+++ b/meson.build
@@ -157,18 +157,6 @@ endif
 add_project_arguments('-DSYSCONFDIR="/@0@"'.format(join_paths(prefix, sysconfdir)), language : 'c')
 
 version = '"@0@"'.format(meson.project_version())
-git = find_program('git', native: true, required: false)
-if git.found()
-	git_commit = run_command([git, '--git-dir=.git', 'rev-parse', '--short', 'HEAD'], check: false)
-	git_branch = run_command([git, '--git-dir=.git', 'rev-parse', '--abbrev-ref', 'HEAD'], check: false)
-	if git_commit.returncode() == 0 and git_branch.returncode() == 0
-		version = '"@0@-@1@ (" __DATE__ ", branch \'@2@\')"'.format(
-			meson.project_version(),
-			git_commit.stdout().strip(),
-			git_branch.stdout().strip(),
-		)
-	endif
-endif
 add_project_arguments('-DSWAY_VERSION=@0@'.format(version), language: 'c')
 
 fs = import('fs')
-- 
2.53.0

