From f96005d8a15eae38b335acac8566b796a5f8a5ea Mon Sep 17 00:00:00 2001
From: huangjf <hjf@t-chip.com.cn>
Date: Wed, 18 May 2022 09:42:35 +0800
Subject: [PATCH] Revert: gpt: return 1 directly when test the mbr sector

---
 disk/part_efi.c | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/disk/part_efi.c b/disk/part_efi.c
index 27f3e4dda3..dadd12ecfb 100644
--- a/disk/part_efi.c
+++ b/disk/part_efi.c
@@ -1083,18 +1083,6 @@ static int is_pmbr_valid(legacy_mbr * mbr)
 {
 	int i = 0;
 
-#ifdef CONFIG_ARCH_ROCKCHIP
-	/*
-	 * In sd-update card, we use RKPARM partition in bootloader to load
-	 * firmware, and use MS-DOS partition in recovery to update system.
-	 * Now, we want to use gpt in bootloader and abandon the RKPARM
-	 * partition. So in new sd-update card, we write the MS-DOS partition
-	 * table and gpt to sd card. Then we must return 1 directly when test
-	 * the mbr sector otherwise the gpt is unavailable.
-	 */
-	return 1;
-#endif
-
 	if (!mbr || le16_to_cpu(mbr->signature) != MSDOS_MBR_SIGNATURE)
 		return 0;
 
-- 
2.17.1

