--- a/bcmdhd.1.363.59.144.x.cn/dhd_config.c	2019-08-10 09:56:36.681753000 +0200
+++ b/bcmdhd.1.363.59.144.x.cn/dhd_config.c	2019-08-10 09:57:00.305559000 +0200
@@ -54,6 +54,7 @@
 #define BCM43362A2_CONF_NAME "config_40181a2.txt"
 #define BCM43438A0_CONF_NAME "config_43438a0.txt"
 #define BCM43438A1_CONF_NAME "config_43438a1.txt"
+#define BCM43438A2_CONF_NAME "config_43436b0.txt"
 #define BCM4334B1_CONF_NAME "config_4334b1.txt"
 #define BCM43341B0_CONF_NAME "config_43341b0.txt"
 #define BCM43241B4_CONF_NAME "config_43241b4.txt"
@@ -125,6 +126,14 @@
 	"fw_bcm43438a1_mfg.bin"
 };
 
+const static char *bcm43436b0_fw_name[] = {
+	"fw_bcm43436b0.bin",
+	"fw_bcm43436b0_apsta.bin",
+	"fw_bcm43436b0_p2p.bin",
+	"fw_bcm43436b0_es.bin",
+	"fw_bcm43436b0_mfg.bin"
+};
+
 const static char *bcm43341b0_ag_fw_name[] = {
 	"fw_bcm43341b0_ag.bin",
 	"fw_bcm43341b0_ag_apsta.bin",
@@ -520,6 +529,8 @@
 				strcpy(&fw_path[i+1], bcm43438a0_fw_name[fw_type]);
 			else if (chiprev == BCM43430A1_CHIP_REV)
 				strcpy(&fw_path[i+1], bcm43438a1_fw_name[fw_type]);
+			else if (chiprev == BCM43430A2_CHIP_REV)
+				strcpy(&fw_path[i+1], bcm43436b0_fw_name[fw_type]);
 			break;
 		case BCM4334_CHIP_ID:
 			if (chiprev == BCM4334B1_CHIP_REV)
@@ -688,6 +699,8 @@
 				strcpy(&conf_path[i+1], BCM43438A0_CONF_NAME);
 			else if (chiprev == BCM43430A1_CHIP_REV)
 				strcpy(&conf_path[i+1], BCM43438A1_CONF_NAME);
+			else if (chiprev == BCM43430A2_CHIP_REV)
+				strcpy(&conf_path[i+1], BCM43436B0_CONF_NAME);
 			break;
 		case BCM4334_CHIP_ID:
 			if (chiprev == BCM4334B1_CHIP_REV)
--- a/bcmdhd.1.363.59.144.x.cn/dhd_config.h	2019-08-10 10:13:00.229280999 +0200
+++ b/bcmdhd.1.363.59.144.x.cn/dhd_config.h	2019-08-10 10:16:16.383309000 +0200
@@ -23,6 +23,7 @@
 #define BCM43362A2_CHIP_REV     1
 #define BCM43430A0_CHIP_REV     0
 #define BCM43430A1_CHIP_REV     1
+#define BCM43430A2_CHIP_REV     2
 #define BCM4330B2_CHIP_REV      4
 #define BCM4334B1_CHIP_REV      3
 #define BCM43341B0_CHIP_REV     2
