diff --git a/Makefile b/Makefile
index 65d0a5c..9f665ba 100755
--- a/Makefile
+++ b/Makefile
@@ -184,7 +184,7 @@ THREADX:
 	$(MAKE) -C $(RT28xx_DIR)/os/Threadx -f $(RT28xx_DIR)/os/ThreadX/Makefile
 
 LINUX:
-ifneq (,$(findstring 2.4,$(LINUX_SRC)))
+ifneq (,)
 
 ifeq ($(OSABL),YES)
 	cp -f os/linux/Makefile.4.util $(RT28xx_DIR)/os/linux/Makefile
@@ -330,7 +330,7 @@ endif
 
 uninstall:
 ifeq ($(TARGET), LINUX)
-ifneq (,$(findstring 2.4,$(LINUX_SRC)))
+ifneq (,)
 	$(MAKE) -C $(RT28xx_DIR)/os/linux -f Makefile.4 uninstall
 else
 	$(MAKE) -C $(RT28xx_DIR)/os/linux -f Makefile.6 uninstall
@@ -339,7 +339,7 @@ endif
 
 install:
 ifeq ($(TARGET), LINUX)
-ifneq (,$(findstring 2.4,$(LINUX_SRC)))
+ifneq (,)
 	$(MAKE) -C $(RT28xx_DIR)/os/linux -f Makefile.4 install
 else
 	$(MAKE) -C $(RT28xx_DIR)/os/linux -f Makefile.6 install
@@ -347,7 +347,7 @@ endif
 endif
 
 libwapi:
-ifneq (,$(findstring 2.4,$(LINUX_SRC)))
+ifneq (,)
 	cp -f os/linux/Makefile.libwapi.4 $(RT28xx_DIR)/os/linux/Makefile
 	$(MAKE) -C $(RT28xx_DIR)/os/linux/
 else
@@ -357,7 +357,7 @@ endif
 
 osutil:
 ifeq ($(OSABL),YES)
-ifneq (,$(findstring 2.4,$(LINUX_SRC)))
+ifneq (,)
 	cp -f os/linux/Makefile.4.util $(RT28xx_DIR)/os/linux/Makefile
 	$(MAKE) -C $(RT28xx_DIR)/os/linux/
 else
@@ -371,7 +371,7 @@ endif
 
 osnet:
 ifeq ($(OSABL),YES)
-ifneq (,$(findstring 2.4,$(LINUX_SRC)))
+ifneq (,)
 	cp -f os/linux/Makefile.4.netif $(RT28xx_DIR)/os/linux/Makefile
 	$(MAKE) -C $(RT28xx_DIR)/os/linux/
 else
@@ -384,7 +384,7 @@ endif
 endif
 
 osdrv:
-ifneq (,$(findstring 2.4,$(LINUX_SRC)))
+ifneq (,)
 	cp -f os/linux/Makefile.4 $(RT28xx_DIR)/os/linux/Makefile
 	$(MAKE) -C $(RT28xx_DIR)/os/linux/
 else
