--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -93,11 +93,11 @@
 
 # Initialize the library directory name suffix.
 if (NOT MINGW AND NOT MSVC AND NOT CMAKE_SYSTEM_NAME MATCHES "FreeBSD|DragonFly")
-if ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
-  set ( _init_lib_suffix "64" )
-else ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+#if ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+#  set ( _init_lib_suffix "64" )
+#else ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
   set ( _init_lib_suffix "" )
-endif ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
+#endif ( CMAKE_SIZEOF_VOID_P EQUAL 8 )
 else ()
   set ( _init_lib_suffix "" )
 endif()

