--- a/es-app/src/guis/GuiMenu.cpp
+++ b/es-app/src/guis/GuiMenu.cpp
@@ -2629,7 +2629,6 @@
 {
 	auto s = new GuiSettings(mWindow, _("SOUND SETTINGS").c_str());
 
-#ifndef _ENABLEEMUELEC
 	s->addGroup(_("VOLUME"));
 
 	// volume
@@ -2647,7 +2646,6 @@
 	volumePopup->setState(Settings::getInstance()->getBool("VolumePopup"));
 	s->addWithLabel(_("SHOW OVERLAY WHEN VOLUME CHANGES"), volumePopup);
 	s->addSaveFunc([volumePopup] { Settings::getInstance()->setBool("VolumePopup", volumePopup->getState()); });
-#endif
 
 	s->addGroup(_("MUSIC"));
 
--- a/es-core/src/Settings.cpp
+++ b/es-core/src/Settings.cpp
@@ -181,7 +181,7 @@
 
 	// Audio out device for volume control
 	#if defined _RPI_ || defined _ENABLEEMUELEC
-		mStringMap["AudioDevice"] = "PCM";
+		mStringMap["AudioDevice"] = "Playback";
 	#else
 		mStringMap["AudioDevice"] = "Master";
 	#endif
