--- a/Utilities/cmcurl/lib/vtls/openssl.c.orig	2016-10-19 19:17:49.615923691 +0200
+++ b/Utilities/cmcurl/lib/vtls/openssl.c	2016-10-19 19:22:50.849565684 +0200
@@ -164,6 +164,11 @@
 #define HAVE_SSL_COMP_FREE_COMPRESSION_METHODS 1
 #endif
 
+#if !defined(OPENSSL_NO_COMP) || \
+  OPENSSL_VERSION_NUMBER >= 0x10100000L /* 1.1.0+ has no free compression methods */ 
+#undef HAVE_SSL_COMP_FREE_COMPRESSION_METHODS /* undef */
+#endif
+
 #ifdef HAVE_OPENSSL3
 #define HAVE_EVP_PKEY_GET_PARAMS 1
 #endif
