From 70c39c72b06564d13675447779fc1595311e227a Mon Sep 17 00:00:00 2001
From: Jérôme Benoit <jerome.benoit@piment-noir.org>
Date: Mon, 24 Oct 2016 19:39:41 +0200
Subject: [PATCH] disable free comp methods

---
diff --git a/Utilities/cmcurl/lib/vtls/openssl.c b/Utilities/cmcurl/lib/vtls/openssl.c
index 4629ca4..cea04b9 100644
--- a/Utilities/cmcurl/lib/vtls/openssl.c
+++ b/Utilities/cmcurl/lib/vtls/openssl.c
@@ -113,6 +113,11 @@
 #include <openssl/ui.h>
 #endif
 
+#if !defined(OPENSSL_NO_COMP) || \
+  OPENSSL_VERSION_NUMBER >= 0x10100000L /* 1.1.0+ has no free compression methods */ 
+#undef HAVE_SSL_COMP_FREE_COMPRESSION_METHODS /* undef */
+#endif
+
 #ifdef HAVE_OPENSSL3
 #define HAVE_EVP_PKEY_GET_PARAMS 1
 #endif
-- 
2.53.0

