--- a/configure.ac	2025-02-26 20:29:12.903050209 +0000
+++ b/configure.ac	2025-02-26 20:29:06.246129502 +0000
@@ -499,7 +499,7 @@
 # If issues found with using a different gettext version than below and
 # performing a crossbuild, update AM_GNU_GETTEXT_VERSION to match and
 # run autogen.sh to update necessary files.
-AM_GNU_GETTEXT_VERSION([0.21])
+AM_GNU_GETTEXT_VERSION([0.24])
 AM_GNU_GETTEXT([external], [need-ngettext])
 LIBS="$LTLIBINTL $LIBS"
 if test "x${prefix}" = xNONE; then
