From f0d5f5f43f05f6c8aea48c7765fd58a735733045 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Wed, 26 Feb 2025 10:39:03 +0000
Subject: [PATCH] gettext

diff --git a/configure.ac b/configure.ac
index 7ffc0a4..3b9c11d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -502,7 +502,7 @@ dnl
 # If issues found with using a different gettext version than below and
 # performing a crossbuild, update AM_GNU_GETTEXT_VERSION to match and
 # run autogen.sh to update necessary files.
-AM_GNU_GETTEXT_VERSION([0.21])
+AM_GNU_GETTEXT_VERSION([0.24])
 AM_GNU_GETTEXT([external], [need-ngettext])
 LIBS="$LTLIBINTL $LIBS"
 if test "x${prefix}" = xNONE; then
-- 
2.53.0

