--- a/Makefile.am
+++ b/Makefile.am
@@ -22,8 +22,6 @@ SUBDIRS = \
 	src \
 	$(proxy_dirs) \
 	contrib \
-	examples \
-	test \
 	po \
 	doc
 
--- a/Makefile.in
+++ b/Makefile.in
@@ -78,7 +78,7 @@ pkgconfigDATA_INSTALL = $(INSTALL_DATA)
   unique=`for i in $$list; do \
     if test -f "$$i"; then echo $$i; else echo $(srcdir)/$$i; fi; \
   done | $(am__uniquify_input)`
-DIST_SUBDIRS = m4 src daemon contrib examples test po doc
+DIST_SUBDIRS = m4 src daemon contrib po doc
 am__DIST_COMMON = $(srcdir)/Makefile.in $(srcdir)/config.h.in \
 	$(srcdir)/zvbi-0.2.pc.in $(srcdir)/zvbi.spec.in \
 	$(top_srcdir)/build-aux/compile \
@@ -268,8 +268,6 @@ SUBDIRS = \
 	src \
 	$(proxy_dirs) \
 	contrib \
-	examples \
-	test \
 	po \
 	doc
 
--- a/configure.ac
+++ b/configure.ac
@@ -384,7 +384,6 @@ AM_CONDITIONAL(BUILD_STATIC_LIB, [test "x$enable_static" = xyes])
 AC_OUTPUT([
   Makefile
   contrib/Makefile
-  examples/Makefile
   daemon/Makefile
   daemon/zvbid.init
   doc/Doxyfile
@@ -392,7 +391,6 @@ AC_OUTPUT([
   m4/Makefile
   src/Makefile
   src/dvb/Makefile
-  test/Makefile
   po/Makefile.in
   zvbi.spec
   zvbi-0.2.pc
