--- a/shell/linux/Makefile	2019-01-18 11:46:08.000000000 -0600
+++ a/shell/linux/Makefile	2019-01-18 14:05:17.236571691 -0600
@@ -204,6 +204,10 @@
         # ODROID-C1 & -C1+
         ifneq (,$(findstring odroidc1,$(platform)))
             MFLAGS += -mtune=cortex-a5
+       else ifneq (,$(findstring odroidc2,$(platform)))
+            MFLAGS += -mtune=cortex-a73.cortex-a53
+            #CFLAGS += -D TARGET_NO_OPENMP 
+            USE_GLES := 1
 
         # ODROID-U2, -U3, -X & -X2
         else
