From a6544dee4456e05aecbf9c758a28697686ca48fe Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@dv9000.openelec.tv>
Date: Sat, 18 Apr 2009 22:06:12 +0200
Subject: [PATCH] halt no init

diff --git a/init/halt.c b/init/halt.c
--- a/init/halt.c
+++ b/init/halt.c
@@ -206,7 +206,7 @@ int halt_main(int argc UNUSED_PARAM, cha
 
 	/* Perform action. */
 	rc = 1;
-	if (!(flags & 4)) { /* no -f */
+	if (ENABLE_INIT && !(flags & 4)) { /* no -f */
 //TODO: I tend to think that signalling linuxrc is wrong
 // pity original author didn't comment on it...
 		if (ENABLE_LINUXRC) {
