From 712ce0b82e49bd55b55b3e5a5db6bc642390d7a1 Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Mon, 11 Sep 2017 20:19:21 -0700
Subject: [PATCH] update shadow or passwd not both

diff --git a/loginutils/passwd.c b/loginutils/passwd.c
--- a/loginutils/passwd.c
+++ b/loginutils/passwd.c
@@ -218,7 +218,7 @@ int passwd_main(int argc UNUSED_PARAM, c
 	if (rc > 0)
 		/* password in /etc/shadow was updated */
 		newp = (char*) "x";
-	if (rc >= 0)
+	if (rc == 0)
 		/* 0 = /etc/shadow missing (not an error), >0 = passwd changed in /etc/shadow */
 #endif
 	{
