From 5d26964a7c0553e922b53105a8a3e5636d586a30 Mon Sep 17 00:00:00 2001
From: Stephan Raue <stephan@openelec.tv>
Date: Wed, 18 Feb 2015 20:08:26 +0100
Subject: [PATCH] dont build fbrun

diff --git a/Makefile.am b/Makefile.am
--- a/Makefile.am
+++ b/Makefile.am
@@ -79,7 +79,6 @@
 include src/Makemodule.am
 include src/tests/Makemodule.am
 include util/Makemodule.am
-include util/fbrun/Makemodule.am
 
 uninstall-hook:
 	rmdir $(DESTDIR)$(pkgdatadir)
