--- a/Core/System.cpp
+++ b/Core/System.cpp
@@ -599,13 +599,13 @@
 	case DIRECTORY_PSP:
 		return pspDirectory;
 	case DIRECTORY_CHEATS:
-		return pspDirectory / "Cheats";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/Cheats/");
 	case DIRECTORY_GAME:
-		return pspDirectory / "GAME";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/GAME/");
 	case DIRECTORY_SAVEDATA:
-		return pspDirectory / "SAVEDATA";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/SAVEDATA");
 	case DIRECTORY_SCREENSHOT:
-		return pspDirectory / "SCREENSHOT";
+		return Path("/storage/roms/screenshots");
 	case DIRECTORY_SYSTEM:
 		return pspDirectory / "SYSTEM";
 	case DIRECTORY_PAUTH:
@@ -615,7 +615,7 @@
 	case DIRECTORY_DUMP:
 		return pspDirectory / "SYSTEM/DUMP";
 	case DIRECTORY_SAVESTATE:
-		return pspDirectory / "PPSSPP_STATE";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/PPSSPP_STATE/");
 	case DIRECTORY_CACHE:
 		return pspDirectory / "SYSTEM/CACHE";
 	case DIRECTORY_TEXTURES:
@@ -630,7 +630,7 @@
 	case DIRECTORY_VIDEO:
 		return pspDirectory / "VIDEO";
 	case DIRECTORY_AUDIO:
-		return pspDirectory / "AUDIO";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/AUDIO/");
 	case DIRECTORY_CUSTOM_SHADERS:
 		return pspDirectory / "shaders";
 
