--- a/Core/System.cpp
+++ b/Core/System.cpp
@@ -592,13 +592,13 @@
 
 	switch (directoryType) {
 	case DIRECTORY_CHEATS:
-		return pspDirectory / "Cheats";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/Cheats/");
 	case DIRECTORY_GAME:
-		return pspDirectory / "GAME";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/GAME/");
 	case DIRECTORY_SAVEDATA:
-		return pspDirectory / "SAVEDATA";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/SAVEDATA");
 	case DIRECTORY_SCREENSHOT:
-		return pspDirectory / "SCREENSHOT";
+		return Path("/storage/roms/screenshots");
 	case DIRECTORY_SYSTEM:
 		return pspDirectory / "SYSTEM";
 	case DIRECTORY_PAUTH:
@@ -606,7 +606,7 @@
 	case DIRECTORY_DUMP:
 		return pspDirectory / "SYSTEM/DUMP";
 	case DIRECTORY_SAVESTATE:
-		return pspDirectory / "PPSSPP_STATE";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/PPSSPP_STATE/");
 	case DIRECTORY_CACHE:
 		return pspDirectory / "SYSTEM/CACHE";
 	case DIRECTORY_TEXTURES:
@@ -621,7 +621,7 @@
 	case DIRECTORY_VIDEO:
 		return pspDirectory / "VIDEO";
 	case DIRECTORY_AUDIO:
-		return pspDirectory / "AUDIO";
+		return Path("/storage/roms/savestates/PPSSPPSDL/PSP/AUDIO/");
 	case DIRECTORY_MEMSTICK_ROOT:
 		return g_Config.memStickDirectory;
 	// Just return the memory stick root if we run into some sort of problem.

