From 63834aaf93d66ea0044ba5dd887805520fe27f6d Mon Sep 17 00:00:00 2001
From: portisch <hugo.portisch@yahoo.de>
Date: Thu, 13 Jun 2024 13:20:42 +0200
Subject: [PATCH] seek: limit seek position to title duration

---
 src/libbluray/bluray.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/libbluray/bluray.c b/src/libbluray/bluray.c
index b3b55f7..3f63d09 100644
--- a/src/libbluray/bluray.c
+++ b/src/libbluray/bluray.c
@@ -1722,8 +1722,9 @@ int64_t bd_seek_time(BLURAY *bd, uint64_t tick)
 
     bd_mutex_lock(&bd->mutex);
 
-    if (bd->title &&
-        tick < bd->title->duration) {
+    if (bd->title) {
+        if (tick > bd->title->duration)
+            tick = bd->title->duration;
 
         _change_angle(bd);
 
-- 
2.45.2

