From e71ade56d0ab2faa257a440af99c2e42feec0ffc Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Sat, 09 Oct 2021 23:35:44 +0200
Subject: [PATCH] cross compile fix

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,7 @@ SOURCES := \
   src/*/*/*.cpp
 
 CPUFLAGS := $(MAKE_CPUFLAGS)
-STRIPCMD := $(or $(STRIP),strip) --strip-all
+STRIPCMD := ${STRIP}
 ifneq ($(ISWIN),)
   OUTNAME := dosbox_pure_libretro.dll
   CXX     ?= g++
@@ -152,6 +152,9 @@ else
   CXX     ?= g++
   LDFLAGS := -Wl,--gc-sections -fno-ident
   COMMONFLAGS += -pthread
+ifeq (1,1) # always
+  CPUFLAGS :=
+else
   ifeq ($(CPUFLAGS),)
     # ARM optimizations
     PROCCPU := $(shell cat /proc/cpuinfo))
@@ -180,6 +183,7 @@ else
     CXX := g++-9
   endif
 endif
+endif
 
 ifeq ($(BUILD),DEBUG)
   BUILDDIR := debug
