From 115fe45bc91b86b54c61de5e5d230484e6fc8901 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 03 Apr 2015 12:19:38 +0300
Subject: [PATCH] m4 tests

---
diff --git a/libsidplay/unix/my_macros.m4 b/libsidplay/unix/my_macros.m4
--- a/libsidplay/unix/my_macros.m4
+++ b/libsidplay/unix/my_macros.m4
@@ -80,8 +80,8 @@
     AC_CACHE_VAL(test_cv_have_ios_binary,
     [
         AC_TRY_COMPILE(
-            [#include <fstream.h>],
-            [ifstream myTest(ios::in|ios::binary);],
+            [#include <fstream>],
+            [std::ifstream myTest(std::ios::in|std::ios::binary);],
             [test_cv_have_ios_binary=yes],
             [test_cv_have_ios_binary=no]
         )
@@ -105,9 +105,9 @@
     AC_CACHE_VAL(test_cv_have_ios_openmode,
     [
         AC_TRY_COMPILE(
-            [#include <fstream.h>
-             #include <iomanip.h>],
-            [ios::openmode myTest = ios::in;],
+            [#include <fstream>
+             #include <iomanip>],
+            [std::ios_base::openmode myTest = std::ios::in;],
             [test_cv_have_ios_openmode=yes],
             [test_cv_have_ios_openmode=no]
         )
