From 115fe45bc91b86b54c61de5e5d230484e6fc8901 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 03 Apr 2015 12:19:38 +0300
Subject: [PATCH] includes

---
diff --git a/builders/hardsid-builder/src/hardsid-builder.cpp b/builders/hardsid-builder/src/hardsid-builder.cpp
--- a/builders/hardsid-builder/src/hardsid-builder.cpp
+++ b/builders/hardsid-builder/src/hardsid-builder.cpp
@@ -55,6 +55,7 @@
 
 #include "hardsid.h"
 #include "hardsid-emu.h"
+#include <string.h>
 
 
 #ifdef HAVE_MSWINDOWS
diff --git a/builders/resid-builder/src/resid.cpp b/builders/resid-builder/src/resid.cpp
--- a/builders/resid-builder/src/resid.cpp
+++ b/builders/resid-builder/src/resid.cpp
@@ -24,6 +24,7 @@
 
 #include "resid.h"
 #include "resid-emu.h"
+#include <string.h>
 
 
 char ReSID::m_credit[];
diff --git a/builders/resid-builder/src/resid-builder.cpp b/builders/resid-builder/src/resid-builder.cpp
--- a/builders/resid-builder/src/resid-builder.cpp
+++ b/builders/resid-builder/src/resid-builder.cpp
@@ -45,6 +45,7 @@
 
 #include "resid.h"
 #include "resid-emu.h"
+#include <string.h>
 
 // Error String(s)
 const char *ReSIDBuilder::ERR_FILTER_DEFINITION = "RESID ERROR: Filter definition is not valid (see docs).";
