From aa44997bf59e2aca86a38cb65755861c413b2609 Mon Sep 17 00:00:00 2001
From: James Le Cuirot <chewi@gentoo.org>
Date: Sun, 18 Feb 2024 09:18:31 +0000
Subject: [PATCH 13/14] v4l2_req: Fix building against musl by including
 pthread.h

---
 libavcodec/v4l2_req_decode_q.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libavcodec/v4l2_req_decode_q.h b/libavcodec/v4l2_req_decode_q.h
index af7bbe1de4..27eafbc42b 100644
--- a/libavcodec/v4l2_req_decode_q.h
+++ b/libavcodec/v4l2_req_decode_q.h
@@ -1,6 +1,8 @@
 #ifndef AVCODEC_V4L2_REQ_DECODE_Q_H
 #define AVCODEC_V4L2_REQ_DECODE_Q_H
 
+#include <pthread.h>
+
 typedef struct req_decode_ent {
     struct req_decode_ent * next;
     struct req_decode_ent * prev;
-- 
2.34.1

