From e00b19de388fec567e37c10edeafa646d322afa6 Mon Sep 17 00:00:00 2001
From: John Cox <jc@kynesim.co.uk>
Date: Thu, 1 Feb 2024 18:11:06 +0000
Subject: [PATCH 08/14] matroskaenc: Fix H264 delayed extradata creation

---
 libavformat/matroskaenc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavformat/matroskaenc.c b/libavformat/matroskaenc.c
index 61e4c976ef..a6a00f03e7 100644
--- a/libavformat/matroskaenc.c
+++ b/libavformat/matroskaenc.c
@@ -1125,7 +1125,7 @@ static int mkv_assemble_native_codecprivate(AVFormatContext *s, AVIOContext *dyn
     case AV_CODEC_ID_WAVPACK:
         return put_wv_codecpriv(dyn_cp, extradata, extradata_size);
     case AV_CODEC_ID_H264:
-        if (par->extradata_size)
+        if (extradata_size)
             return ff_isom_write_avcc(dyn_cp, extradata,
                                       extradata_size);
         else
-- 
2.34.1

