--- a/bootstrap.conf	2021-03-12 15:09:51.000000000 +0000
+++ b/bootstrap.conf	2021-04-19 13:22:41.000000000 +0000
@@ -16,7 +16,7 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
 
-GNULIB_REVISION=9f48fb992a3d7e96610c4ce8be969cff2d61a01b
+GNULIB_REVISION=3198293e3fd04182f457cb76ff8c5b4ac1155103
 
 # gnulib modules used by this package.
 # mbswidth is used by fix-width.diff's changes to argp rather than directly.
