--- a/qtbase/src/corelib/global/qendian.h
+++ b/qtbase/src/corelib/global/qendian.h
@@ -47,6 +47,8 @@
 // include stdlib.h and hope that it defines __GLIBC__ for glibc-based systems
 #include <stdlib.h>
 #include <string.h>
+#include <stdexcept>
+#include <limits>
 
 #ifdef min // MSVC
 #undef min

--- a/qtbase/src/corelib/tools/qbytearraymatcher.h
+++ b/qtbase/src/corelib/tools/qbytearraymatcher.h
@@ -41,6 +41,8 @@
 #define QBYTEARRAYMATCHER_H
 
 #include <QtCore/qbytearray.h>
+#include <stdexcept>
+#include <limits>
 
 QT_BEGIN_NAMESPACE
 
--- a/qtbase/src/tools/moc/generator.cpp
+++ b/qtbase/src/tools/moc/generator.cpp
@@ -42,6 +42,8 @@
 
 #include <math.h>
 #include <stdio.h>
+#include <stdexcept>
+#include <limits>
 
 #include <private/qmetaobject_p.h> //for the flags.
 #include <private/qplugin_p.h> //for the flags.

