From 01bf6480e8a0a13089f41ed28a9055295066baa0 Mon Sep 17 00:00:00 2001
From: Stefan Saraev <stefan@saraev.ca>
Date: Fri, 09 Jan 2015 21:25:29 +0200
Subject: [PATCH] do not cleanup routes

diff --git a/src/device.c b/src/device.c
index 0fda950..eb09e53 100644
--- a/src/device.c
+++ b/src/device.c
@@ -1501,8 +1501,6 @@ int __connman_device_init(const char *de
 	if (nodevice)
 		nodevice_filter = g_strsplit(nodevice, ",", -1);
 
-	cleanup_devices();
-
 	return 0;
 }
 
