diff -Naur a/linux/Makefile b/linux/Makefile
--- a/linux/Makefile
+++ b/linux/Makefile
@@ -112,6 +112,19 @@
 
 untar: linux-media.tar.bz2
 	tar xfj linux-media.tar.bz2
+	mkdir -p drivers/media/amlogic/
+	# Copy amlvideodri module
+	cp -a "$(SRCDIR)/drivers/amlogic/video_dev" "drivers/media/amlogic"
+	sed -i 's,common/,,g; s,"trace/,",g' `find drivers/media/amlogic/video_dev/ -type f`
+	# Copy videobuf-res module
+	cp -a "$(SRCDIR)/drivers/media/v4l2-core/videobuf-res.c" "drivers/media/v4l2-core/"
+	cp -a "$(SRCDIR)/include/media/videobuf-res.h" "include/media/"
+
+	# Copy WeTek Play DVB driver
+	cp -a "$(SRCDIR)/drivers/amlogic/wetek" "drivers/media/amlogic"
+	# fix includes
+	sed -e 's/#include "dvb_frontend.h"/#include "media\/dvb_frontend.h"/g' -i drivers/media/amlogic/wetek/*.*
+	sed -e 's/#include "dvb_math.h"/#include "media\/dvb_math.h"/g' -i drivers/media/amlogic/wetek/*.*
 	-rm -f .patches_applied .linked_dir .git_log.md5
 
 clean:
