--- a/Makefile
+++ b/Makefile
@@ -359,6 +359,30 @@
 	PLATFORM_EXT := unix
 	WITH_DYNAREC=arm
 	HAVE_GENERIC_JIT = 0
+
+# RockPro64
+
+else ifeq ($(platform), amlogicng)
+	EXT ?= so
+	TARGET := $(TARGET_NAME)_libretro.$(EXT)
+	SHARED += -shared -Wl,--version-script=link.T
+	fpic = -fPIC
+	LIBS += -lrt
+	ARM_FLOAT_ABI_HARD = 1
+	FORCE_GLES = 1
+	SINGLE_PREC_FLAGS = 1
+	CPUFLAGS += -DNO_ASM -DARM_ASM -frename-registers -ftree-vectorize
+	CFLAGS += -marm -mfloat-abi=hard -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53 -mfpu=neon-fp-armv8 $(CPUFLAGS)
+	CXXFLAGS += -marm -mfloat-abi=hard -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53 -mfpu=neon-fp-armv8 $(CPUFLAGS)
+	ifeq ($(HAVE_CLANG),0)
+		CFLAGS += -mvectorize-with-neon-quad
+		CXXFLAGS += -mvectorize-with-neon-quad
+	endif
+	ASFLAGS += $(CFLAGS) -c -frename-registers -fno-strict-aliasing -ffast-math -ftree-vectorize
+	PLATFORM_EXT := unix
+	WITH_DYNAREC=arm
+	HAVE_GENERIC_JIT = 0
+
 
 # Tinkerboard
 else ifeq ($(platform), tinkerboard)
