From 41e6173800b1cb9bc820506363cc1bd51fc3f16d Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Mon, 25 Apr 2016 10:37:14 -0700
Subject: [PATCH] crosscompile

diff --git a/Makefile b/Makefile
index f98abbd..9d1d524 100755
--- a/Makefile
+++ b/Makefile
@@ -38,13 +38,13 @@ EXEC=hid_mapper
 VERSION=2.1.0
 
 all: $(OBJS)
-	g++ $(LDFLAGS) $(OBJS) -o $(EXEC)
+	${CXX} $(LDFLAGS) $(OBJS) -o $(EXEC)
 
 %.o: %.cpp
-	g++ -c $(CPPFLAGS) $<
+	${CXX} -c $(CPPFLAGS) $<
 
 %.o: %.c
-	gcc -c $(CFLAGS) $<
+	${CC} -c $(CFLAGS) $<
 
 deb: all
 	rm -rf package/
