--- a/Makefile
+++ b/Makefile
@@ -163,18 +163,18 @@
 endif
 
 RM     = rm -f
-AS     = as
-CC     = gcc
-CXX    = g++
-STRIP  ?= strip
+#AS     = as
+#CC     = gcc
+#CXX    = g++
+#STRIP  ?= strip
 PROG   = amiberry
 
 #
 # SDL2 options
 #
 all: guisan $(PROG)
-export SDL_CFLAGS := $(shell sdl2-config --cflags)
-export SDL_LDFLAGS := $(shell sdl2-config --libs)
+export SDL_CFLAGS := $(shell $(SYSROOT_PREFIX)/usr/bin/sdl2-config --cflags)
+export SDL_LDFLAGS := $(shell $(SYSROOT_PREFIX)/usr/bin/sdl2-config --libs)
 
 CPPFLAGS += $(SDL_CFLAGS) -Iguisan-dev/include
 LDFLAGS += $(SDL_LDFLAGS) -lSDL2_image -lSDL2_ttf -lguisan -Lguisan-dev/lib
@@ -223,7 +223,7 @@
     CFLAGS += -fsanitize=leak -fsanitize-recover=address
 endif
 
-LDFLAGS += -lpthread -lz -lpng -lrt -lxml2 -lFLAC -lmpg123 -ldl -lmpeg2convert -lmpeg2
+LDFLAGS += -lpthread -lz -lpng -lrt -lxml2 -lFLAC -lmpg123 -ldl -lmpeg2convert -lmpeg2 -logg
 export ASFLAGS = $(CPUFLAGS)
 
 export CFLAGS := $(CPUFLAGS) $(CFLAGS) -pipe -Wno-shift-overflow -Wno-narrowing $(EXTRA_CFLAGS)
