From 8227a01fc92d0f1d78731c8da377f43f117fa863 Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Mon, 09 Dec 2024 13:09:43 +0000
Subject: [PATCH] fix OutputFrame declaration for gcc15

diff --git a/comskip.c b/comskip.c
index 7d1b8e0..dbdace6 100644
--- a/comskip.c
+++ b/comskip.c
@@ -978,7 +978,7 @@ int					FindBlackThreshold(double percentile);
 int					FindUniformThreshold(double percentile);
 void				OutputFrameArray(bool screenOnly);
 void                OutputBlackArray();
-void				OutputFrame();
+void				OutputFrame(int frame_number);
 void				OpenOutputFiles();
 void				InitializeFrameArray(long i);
 void				InitializeBlackArray(long i);
-- 
2.53.0

