--- a/projects/unix/Makefile
+++ b/projects/unix/Makefile
@@ -295,7 +295,7 @@
 
 # test for presence of SDL
 ifeq ($(origin SDL_CFLAGS) $(origin SDL_LDLIBS), undefined undefined)
-  SDL_CONFIG = $(CROSS_COMPILE)sdl2-config
+  SDL_CONFIG = $(SYSROOT_PREFIX)/usr/bin/sdl2-config
   ifeq ($(shell which $(SDL_CONFIG) 2>/dev/null),)
     SDL_CONFIG = $(CROSS_COMPILE)sdl-config
     ifeq ($(shell which $(SDL_CONFIG) 2>/dev/null),)
@@ -371,9 +371,6 @@
 endif
 
 # set base program pointers and flags
-CC        = $(CROSS_COMPILE)gcc
-CXX       = $(CROSS_COMPILE)g++
-STRINGS   = $(CROSS_COMPILE)strings
 AS        = nasm
 TR       ?= tr
 RM       ?= rm -f
@@ -751,7 +748,7 @@
 # Script hackery for generating ASM include files for the new dynarec assembly code
 $(SRCDIR)/asm_defines/asm_defines_gas.h: $(SRCDIR)/asm_defines/asm_defines_nasm.h
 $(SRCDIR)/asm_defines/asm_defines_nasm.h: $(ASM_DEFINES_OBJ) ../../tools/gen_asm_defines.awk
-	$(STRINGS) "$<" | $(TR) -d '\r' | $(AWK) -v dest_dir="$(SRCDIR)/asm_defines" -f ../../tools/gen_asm_defines.awk
+	$(AWK) -v dest_dir="$(SRCDIR)/asm_defines" -f ../../tools/gen_asm_defines.awk "$<"
 
 # standard build rules
 $(OBJDIR)/%.o: $(SRCDIR)/%.asm $(SRCDIR)/asm_defines/asm_defines_nasm.h

