diff -Naur a/giscanner/gdumpparser.py b/giscanner/gdumpparser.py
--- a/giscanner/gdumpparser.py	2017-04-09 23:59:43.000000000 -0700
+++ b/giscanner/gdumpparser.py	2017-09-18 18:54:55.468436047 -0700
@@ -167,6 +167,8 @@
         launcher = os.environ.get('GI_CROSS_LAUNCHER')
         if launcher:
             args.extend(launcher.split())
+            if self._binary.args[0].split('/')[-1] == 'GIRepository-2.0':
+                self._binary.args[0] = self._binary.args[0].replace('GIRepository-2.0', '.libs/GIRepository-2.0')
 
         args.extend(self._binary.args)
         args.append('--introspect-dump=%s,%s' % (in_path, out_path))
