diff -Naur a/common.mk b/common.mk
--- a/common.mk	2017-04-09 23:59:43.000000000 -0700
+++ b/common.mk	2017-09-16 13:37:08.124776788 -0700
@@ -6,7 +6,7 @@
 # module itself.
 #
 
-INTROSPECTION_SCANNER = \
+INTROSPECTION_SCANNER ?= \
     env PATH=".libs:$(PATH)" \
         LPATH=.libs \
         CC="$(CC)" \
@@ -27,7 +27,7 @@
 # GI_CROSS_LAUNCHER is the command to use for executing g-ir-compiler.
 # Normally will be undefined but can be set (e.g. to wine or qemu)
 # when cross-compiling
-INTROSPECTION_COMPILER = \
+INTROSPECTION_COMPILER ?= \
     env PATH=".libs:$(PATH)" \
         $(GI_CROSS_LAUNCHER) \
         $(top_builddir)/g-ir-compiler$(EXEEXT)
diff -Naur a/giscanner/shlibs.py b/giscanner/shlibs.py
--- a/giscanner/shlibs.py	2017-04-09 23:59:43.000000000 -0700
+++ b/giscanner/shlibs.py	2017-09-16 13:13:56.412694976 -0700
@@ -108,7 +108,10 @@
         if platform_system == 'Darwin':
             args.extend(['otool', '-L', binary.args[0]])
         else:
-            args.extend(['ldd', binary.args[0]])
+            ldd = os.getenv('GI_LDD')
+            if not ldd:
+                ldd = 'ldd'
+            args.extend([ldd, binary.args[0]])
         proc = subprocess.Popen(args, stdout=subprocess.PIPE)
         patterns = {}
         for library in libraries:
