--- a//es-app/src/views/ViewController.cpp
+++ b//es-app/src/views/ViewController.cpp
@@ -16,6 +16,9 @@
 #include "Settings.h"
 #include "SystemData.h"
 #include "Window.h"
+#include "guis/GuiDetectDevice.h"
+
+
 
 ViewController* ViewController::sInstance = NULL;
 
@@ -359,6 +362,15 @@
 	if(mLockInput)
 		return true;
 
+        // batocera
+	/* if we receive a button pressure for a non configured joystick, suggest the joystick configuration */
+        if(config->isConfigured() == false) {
+	  if(input.type == TYPE_BUTTON) {
+	    mWindow->pushGui(new GuiDetectDevice(mWindow, false, NULL));
+	    return true;
+	  }
+        }
+
 	// open menu
 	if(!UIModeController::getInstance()->isUIModeKid() && config->isMappedTo("start", input) && input.value != 0)
 	{
