Subject: revert [PATCH] Remove trailing '/' in getGenericPath
https://github.com/RetroPie/EmulationStation/commit/15ccfc5

--- a/es-core/src/utils/FileSystemUtil.cpp
+++ b/es-core/src/utils/FileSystemUtil.cpp
@@ -229,10 +229,6 @@
 			while((offset = path.find("//")) != std::string::npos)
 				path.erase(offset, 1);
 
-			// remove trailing '/'
-			while(path.length() && ((offset = path.find_last_of('/')) == (path.length() - 1)))
-				path.erase(offset, 1);
-
 			// return generic path
 			return path;
 

