diff --git a/setup.py b/setup.py
index 8eb0b3c..277e44a 100755
--- a/setup.py
+++ b/setup.py
@@ -335,7 +335,7 @@
     )
 
     def initialize_options(self):
-        self.disable_platform_guessing = None
+        self.disable_platform_guessing = True
         self.add_imaging_libs = ""
         build_ext.initialize_options(self)
         for x in self.feature:
@@ -987,6 +987,7 @@
 
 try:
     setup(
+        version=PILLOW_VERSION,
         cmdclass={"build_ext": pil_build_ext},
         ext_modules=ext_modules,
         zip_safe=not (debug_build() or PLATFORM_MINGW),
