From 4d75f7a6a829d079462633cbb2aa3f510ba08891 Mon Sep 17 00:00:00 2001
From: Portisch <hugo.portisch@yahoo.de>
Date: Tue, 11 Oct 2022 16:58:45 +0200
Subject: [PATCH 2/2] fix module name 'FlatBuffers'

---
 dependencies/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dependencies/CMakeLists.txt b/dependencies/CMakeLists.txt
index f64b0d38..a9b4daf6 100644
--- a/dependencies/CMakeLists.txt
+++ b/dependencies/CMakeLists.txt
@@ -75,7 +75,7 @@ if(ENABLE_FLATBUF_SERVER OR ENABLE_FLATBUF_CONNECT)
 
 	if (USE_SYSTEM_FLATBUFFERS_LIBS)
 		find_program(FLATBUFFERS_FLATC_EXECUTABLE NAMES flatc REQUIRED)
-		find_package(Flatbuffers REQUIRED)
+		find_package(FlatBuffers REQUIRED)
 	else ()
 		set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared flatbuffers library")
 		set(FLATBUFFERS_BUILD_TESTS OFF CACHE BOOL "Build Flatbuffers with tests")
-- 
2.38.0

