From 621ad28e4a278445e674986c9ddbf589dffaf228 Mon Sep 17 00:00:00 2001
From: Lee Duncan <lduncan@suse.com>
Date: Sat, 2 Dec 2017 14:03:14 -0800
Subject: [PATCH] Fix undefined call to writev(): include <sys/uio.h>

---
 usr/io.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/usr/io.c b/usr/io.c
index 48b233cf6..e4d00ae2c 100644
--- a/usr/io.c
+++ b/usr/io.c
@@ -28,6 +28,7 @@
 #include <sys/ioctl.h>
 #include <netinet/tcp.h>
 #include <arpa/inet.h>
+#include <sys/uio.h>
 
 #include "types.h"
 #include "iscsi_proto.h"
