From 20cb0d64473ce69b50ebbb3b51f6287c0cf558f3 Mon Sep 17 00:00:00 2001
From: thoradia <22841905+thoradia@users.noreply.github.com>
Date: Fri, 23 Oct 2020 09:59:09 +0200
Subject: [PATCH] ldflags

diff --git a/source/config/mh-linux b/source/config/mh-linux
--- a/source/config/mh-linux
+++ b/source/config/mh-linux
@@ -23,7 +23,7 @@
 LD_RPATH_PRE = -Wl,-rpath,
 
 ## These are the library specific LDFLAGS
-LDFLAGSICUDT=-nodefaultlibs -nostdlib
+#LDFLAGSICUDT=-nodefaultlibs -nostdlib
 
 ## Compiler switch to embed a library name
 # The initial tab in the next line is to prevent icu-config from reading it.
