From 71478aed635bfa8f458db4aa56022272a6a8badc Mon Sep 17 00:00:00 2001
From: Lukas Rusak <lorusak@gmail.com>
Date: Mon, 21 Dec 2015 09:12:10 +0000
Subject: [PATCH] remove av7110_loadkeys

---
 util/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/util/Makefile b/util/Makefile
index c78598c..c3365fb 100644
--- a/util/Makefile
+++ b/util/Makefile
@@ -4,7 +4,6 @@
 
 all clean install:
 	$(MAKE) -C atsc_epg $@
-	$(MAKE) -C av7110_loadkeys $@
 	$(MAKE) -C dib3000-watch $@
 	$(MAKE) -C dst-utils $@
 	$(MAKE) -C dvbdate $@
-- 
2.43.0

