From 2a1869250e30748235f1a11d32619bf2a54a29d5 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Thu, 5 Oct 2017 22:04:11 +0200
Subject: [PATCH 4/4] bcmdhd: if driver is not yet initialized, wait and retry

---
 bcmdhd.1.363.59.144.x.cn/dhd_linux.c | 15 ++++++++++++---
 1 file changed, 12 insertions(+), 3 deletions(-)

diff --git a/bcmdhd.1.363.59.144.x.cn/dhd_linux.c b/bcmdhd.1.363.59.144.x.cn/dhd_linux.c
index c04d185..4f3b45f 100644
--- a/bcmdhd.1.363.59.144.x.cn/dhd_linux.c
+++ b/bcmdhd.1.363.59.144.x.cn/dhd_linux.c
@@ -6167,9 +6167,18 @@ dhd_open(struct net_device *net)
 	wifi_adapter_info_t *adapter = NULL;
 #endif
 
-	if (!dhd_download_fw_on_driverload && !dhd_driver_init_done) {
-		DHD_ERROR(("%s: WLAN driver is not initialized\n", __FUNCTION__));
-		return -1;
+	int retry;
+	for (retry = 0; ++retry; ) {
+		if (!dhd_download_fw_on_driverload && !dhd_driver_init_done) {
+			DHD_ERROR(("%s: WLAN driver is not initialized\n", __FUNCTION__));
+			if (retry > 1) {
+				return -1;
+			} else {
+				OSL_SLEEP(1000);
+			}
+		} else {
+			break;
+		}
 	}
 
 	printf("%s: Enter %p\n", __FUNCTION__, net);
-- 
2.25.1

