From a0ba99d88906161364bdf9277a65f43ec3c5bfaa Mon Sep 17 00:00:00 2001
From: Portisch <hugo.portisch@yahoo.de>
Date: Tue, 8 Dec 2020 09:35:08 -0500
Subject: [PATCH] add-power-on-argument

---
 utils/keytable/keytable.c | 132 +++++++++++++++++++++++---------------
 1 file changed, 81 insertions(+), 51 deletions(-)

diff --git a/utils/keytable/keytable.c b/utils/keytable/keytable.c
index 3df5fcf2..d52bef9b 100644
--- a/utils/keytable/keytable.c
+++ b/utils/keytable/keytable.c
@@ -266,6 +266,7 @@ static const struct argp_option options[] = {
 	{"clear",	'c',	0,		0,	N_("Clears the scancode to keycode mappings"), 0},
 	{"sysdev",	's',	N_("SYSDEV"),	0,	N_("rc device to control, defaults to rc0 if not specified"), 0},
 	{"test",	't',	0,		0,	N_("test if IR is generating events"), 0},
+	{"power-key",	'u',	0,		0,	N_("will scan for NEC power on code"), 0},
 	{"read",	'r',	0,		0,	N_("reads the current scancode/keycode mapping"), 0},
 	{"write",	'w',	N_("KEYMAP"),	0,	N_("write (adds) the keymap from the specified file"), 0},
 	{"set-key",	'k',	N_("SCANKEY"),	0,	N_("Change scan/key pairs"), 0},
@@ -289,6 +290,7 @@ static int readtable = 0;
 static int clear = 0;
 int debug = 0;
 static int test = 0;
+static int poweron = 0;
 static int delay = -1;
 static int period = -1;
 static int test_keymap = 0;
@@ -538,6 +540,9 @@ static error_t parse_opt(int k, char *arg, struct argp_state *state)
 	case 't':
 		test++;
 		break;
+	case 'u':
+		poweron++;
+		break;
 	case 'c':
 		clear++;
 		break;
@@ -1501,7 +1506,7 @@ static void print_scancodes(const struct lirc_scancode *scancodes, unsigned coun
 	}
 }
 
-static void test_event(struct rc_device *rc_dev, int fd)
+static void test_event(struct rc_device *rc_dev, int fd, int power_on)
 {
 	struct input_event ev[64];
 	struct lirc_scancode sc[64];
@@ -1526,7 +1531,11 @@ static void test_event(struct rc_device *rc_dev, int fd)
 		}
 	}
 
-	printf (_("Testing events. Please, press CTRL-C to abort.\n"));
+	if (!power_on)
+		printf (_("Testing events. Please, press CTRL-C to abort.\n"));
+	else
+		printf (_("Waiting for NEC decode event. Please, press CTRL-C to abort.\n"));
+
 	while (1) {
 		struct pollfd pollstruct[2] = {
 			{ .fd = fd, .events = POLLIN },
@@ -1561,54 +1570,72 @@ static void test_event(struct rc_device *rc_dev, int fd)
 			return;
 		}
 
-		for (i = 0; i < rd / sizeof(struct input_event); i++) {
-			printf(_("%ld.%06ld: event type %s(0x%02x)"),
-				ev[i].input_event_sec, ev[i].input_event_usec,
-				get_event_name(events_type, ev[i].type), ev[i].type);
+		if (!power_on) {
+			for (i = 0; i < rd / sizeof(struct input_event); i++) {
+				printf(_("%ld.%06ld: event type %s(0x%02x)"),
+					ev[i].input_event_sec, ev[i].input_event_usec,
+					get_event_name(events_type, ev[i].type), ev[i].type);
 
-			switch (ev[i].type) {
-			case EV_SYN:
-				printf(".\n");
-				break;
-			case EV_KEY:
-				printf(_(" key_%s: %s(0x%04x)\n"),
-					(ev[i].value == 0) ? _("up") : _("down"),
-					get_event_name(key_events, ev[i].code),
-					ev[i].code);
-				break;
-			case EV_REL:
-				printf(_(": %s (0x%04x) value=%d\n"),
-					get_event_name(rel_events, ev[i].code),
-					ev[i].code,
-					ev[i].value);
-				break;
-			case EV_ABS:
-				printf(_(": %s (0x%04x) value=%d\n"),
-					get_event_name(abs_events, ev[i].code),
-					ev[i].code,
-					ev[i].value);
-				break;
-			case EV_MSC:
-				if (ev[i].code == MSC_SCAN)
-					printf(_(": scancode = 0x%02x\n"), ev[i].value);
-				else
-					printf(_(": code = %s(0x%02x), value = %d\n"),
-						get_event_name(msc_events, ev[i].code),
+				switch (ev[i].type) {
+				case EV_SYN:
+					printf(".\n");
+					break;
+				case EV_KEY:
+					printf(_(" key_%s: %s(0x%04x)\n"),
+						(ev[i].value == 0) ? _("up") : _("down"),
+						get_event_name(key_events, ev[i].code),
+						ev[i].code);
+					break;
+				case EV_REL:
+					printf(_(": %s (0x%04x) value=%d\n"),
+						get_event_name(rel_events, ev[i].code),
+						ev[i].code,
+						ev[i].value);
+					break;
+				case EV_ABS:
+					printf(_(": %s (0x%04x) value=%d\n"),
+						get_event_name(abs_events, ev[i].code),
+						ev[i].code,
+						ev[i].value);
+					break;
+				case EV_MSC:
+					if (ev[i].code == MSC_SCAN)
+						printf(_(": scancode = 0x%02x\n"), ev[i].value);
+					else
+						printf(_(": code = %s(0x%02x), value = %d\n"),
+							get_event_name(msc_events, ev[i].code),
+							ev[i].code, ev[i].value);
+					break;
+				case EV_REP:
+					printf(_(": value = %d\n"), ev[i].value);
+					break;
+				case EV_SW:
+				case EV_LED:
+				case EV_SND:
+				case EV_FF:
+				case EV_PWR:
+				case EV_FF_STATUS:
+				default:
+					printf(_(": code = 0x%02x, value = %d\n"),
 						ev[i].code, ev[i].value);
-				break;
-			case EV_REP:
-				printf(_(": value = %d\n"), ev[i].value);
-				break;
-			case EV_SW:
-			case EV_LED:
-			case EV_SND:
-			case EV_FF:
-			case EV_PWR:
-			case EV_FF_STATUS:
-			default:
-				printf(_(": code = 0x%02x, value = %d\n"),
-					ev[i].code, ev[i].value);
-				break;
+					break;
+				}
+			}
+		} else {
+			for (i = 0; i < rd / sizeof(struct input_event); i++) {
+
+				switch (ev[i].type) {
+				case EV_MSC:
+					if (ev[i].code == MSC_SCAN) {
+						unsigned int nec_code =
+								((~ev[i].value & 0xFF) << 24) |
+								((ev[i].value & 0xFF) << 16) |
+								((~(ev[i].value >> 8) & 0xFF) << 8) |
+								((ev[i].value >> 8) & 0xFF);
+						printf(_("Received NEC code: 0x%08X\n"), nec_code);
+					}
+					break;
+				}
 			}
 		}
 	}
@@ -2047,7 +2074,7 @@ int main(int argc, char *argv[])
 		return 0;
 
 	/* Just list all devices */
-	if (!clear && !readtable && !keytable && !ch_proto && !cfg.next && !test && delay < 0 && period < 0 && !bpf_protocol) {
+	if (!clear && !readtable && !keytable && !ch_proto && !cfg.next && !test && !poweron && delay < 0 && period < 0 && !bpf_protocol) {
 		if (show_sysfs_attribs(&rc_dev, devclass))
 			return -1;
 
@@ -2153,6 +2180,9 @@ int main(int argc, char *argv[])
 	/*
 	 * Third step: change protocol
 	 */
+	if (poweron)
+		ch_proto = SYSFS_NEC;
+
 	if (ch_proto || bpf_protocol) {
 		if (rc_dev.lirc_name)
 			clear_bpf(rc_dev.lirc_name);
@@ -2203,8 +2233,8 @@ int main(int argc, char *argv[])
 		set_rate(fd, new_delay, new_period);
 	}
 
-	if (test)
-		test_event(&rc_dev, fd);
+	if (test || poweron)
+		test_event(&rc_dev, fd, poweron);
 
 	return 0;
 }
-- 
2.17.1

