From f92ead39935e8a7e4d799c60def3637b8cb602e8 Mon Sep 17 00:00:00 2001
From: popcornmix <popcornmix@gmail.com>
Date: Thu, 27 Nov 2014 16:31:56 +0000
Subject: [PATCH] [languageinvoker] Reduce priority of python threads

---
 xbmc/interfaces/generic/LanguageInvokerThread.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/xbmc/interfaces/generic/LanguageInvokerThread.cpp b/xbmc/interfaces/generic/LanguageInvokerThread.cpp
index dab0944eee2b..5dba5af2924f 100644
--- a/xbmc/interfaces/generic/LanguageInvokerThread.cpp
+++ b/xbmc/interfaces/generic/LanguageInvokerThread.cpp
@@ -63,8 +63,12 @@ bool CLanguageInvokerThread::execute(const std::string &script, const std::vecto
     m_condition.notify_one();
   }
   else
+  {
     Create();
 
+    /* low prio */
+    SetPriority(GetPriority()-1);
+  }
   //Todo wait until running
 
   return true;
