From 0deafb22dda9aec163eba3f3cc25efc186506af6 Mon Sep 17 00:00:00 2001
From: adamg <adamg88@users.noreply.github.com>
Date: Thu, 23 May 2019 00:12:49 +0100
Subject: [PATCH] Ensure that some add-ons are always enabled at Kodi startup

---
 xbmc/addons/AddonDatabase.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xbmc/addons/AddonDatabase.cpp b/xbmc/addons/AddonDatabase.cpp
index 3e4581f..c0ee74f 100644
--- a/xbmc/addons/AddonDatabase.cpp
+++ b/xbmc/addons/AddonDatabase.cpp
@@ -361,6 +361,10 @@ void CAddonDatabase::SyncInstalled(const std::set<std::string>& ids,
           ORIGIN_SYSTEM, id.c_str()));
     }
 
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID == 'inputstream.adaptive'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID == 'inputstream.rtmp'"));
+    m_pDS->exec(PrepareSQL("UPDATE installed SET enabled=1 WHERE addonID LIKE 'resource.language.%%'"));
+
     CommitTransaction();
   }
   catch (...)
-- 
2.7.4

