From c7a9307565c76fd793a6ee87b0442d37173cc087 Mon Sep 17 00:00:00 2001
From: Kai Sommerfeld <kai.sommerfeld@gmx.com>
Date: Sat, 1 Jun 2019 13:23:22 +0200
Subject: [PATCH] [PVR] Fix deadlock on PVR shutdown.

---
 xbmc/pvr/PVRManager.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/xbmc/pvr/PVRManager.cpp b/xbmc/pvr/PVRManager.cpp
index 6b9ea7ae2b3b..d61a8ec99fef 100644
--- a/xbmc/pvr/PVRManager.cpp
+++ b/xbmc/pvr/PVRManager.cpp
@@ -299,8 +299,6 @@ void CPVRManager::Start()
   // lock here (m_startStopMutex), which only gets hold while starting/restarting pvr manager.
   Stop();
 
-  CSingleLock lock(m_critSection);
-
   if (!m_addons->HasCreatedClients())
     return;
 
@@ -339,8 +337,6 @@ void CPVRManager::Stop(void)
 
   SetState(ManagerStateInterrupted);
 
-  CSingleLock lock(m_critSection);
-
   UnloadComponents();
   m_database->Close();
 

