diff a/xbmc/cores/VideoPlayer/VideoPlayer.cpp b/xbmc/cores/VideoPlayer/VideoPlayer.cpp
--- a/xbmc/cores/VideoPlayer/VideoPlayer.cpp	2019-03-11 12:19:10.843754310 +0000
+++ b/xbmc/cores/VideoPlayer/VideoPlayer.cpp	2019-03-11 12:23:01.105354410 +0000
@@ -1339,7 +1339,6 @@
     {
       if (m_pDemuxer->SeekTime(starttime, true, &startpts))
       {
-        FlushBuffers(starttime / 1000 * AV_TIME_BASE, true, true);
         CLog::Log(LOGDEBUG, "%s - starting demuxer from: %d", __FUNCTION__, starttime);
       }
       else
