From 06ca5f8db5a0c837b245624ba43ffbeca4018b2b Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Tue, 15 Aug 2017 00:06:47 +0200
Subject: [PATCH] EGLNativeTypeAmlogic: turn 3D off for resolution switch

---
 xbmc/windowing/egl/EGLNativeTypeAmlogic.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/xbmc/utils/AMLUtils.cpp b/xbmc/utils/AMLUtils.cpp
index 6ec77ec..01d5942 100644
--- a/xbmc/utils/AMLUtils.cpp
+++ b/xbmc/utils/AMLUtils.cpp
@@ -621,6 +621,7 @@ bool aml_set_native_resolution(const RESOLUTION_INFO &res, std::string framebuff
 {
   bool result = false;
 
+  aml_handle_display_stereo_mode(RENDER_STEREO_MODE_OFF);
   result = aml_set_display_resolution(res, framebuffer_name);
 
   aml_handle_scale(res);
--
2.7.4
