diff --git a/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp b/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp
index 65e66f2..1fda429 100644
--- a/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp
+++ b/xbmc/cores/VideoPlayer/VideoPlayerAudio.cpp
@@ -94,6 +94,9 @@ bool CVideoPlayerAudio::OpenStream(CDVDStreamInfo hints)
     allowpassthrough = false;
 
   CAEStreamInfo::DataType streamType = m_audioSink.GetPassthroughStreamType(hints.codec, hints.samplerate);
+  if (hints.profile == FF_PROFILE_DTS_HD_HRA)
+   streamType = CAEStreamInfo::DataType::STREAM_TYPE_NULL;
+
   CDVDAudioCodec* codec = CDVDFactoryCodec::CreateAudioCodec(hints, m_processInfo,
                                                              allowpassthrough, m_processInfo.AllowDTSHDDecode(),
                                                              streamType);
@@ -601,6 +604,9 @@ bool CVideoPlayerAudio::SwitchCodecIfNeeded()
     allowpassthrough = false;
 
   CAEStreamInfo::DataType streamType = m_audioSink.GetPassthroughStreamType(m_streaminfo.codec, m_streaminfo.samplerate);
+  if (m_streaminfo.profile == FF_PROFILE_DTS_HD_HRA)
+   streamType = CAEStreamInfo::DataType::STREAM_TYPE_NULL;
+
   CDVDAudioCodec *codec = CDVDFactoryCodec::CreateAudioCodec(m_streaminfo, m_processInfo,
                                                              allowpassthrough, m_processInfo.AllowDTSHDDecode(),
                                                              streamType);
