--- a/xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp
+++ b/xbmc/cores/VideoPlayer/DVDInputStreams/DVDInputStreamPVRManager.cpp
@@ -468,7 +468,7 @@
     dStream->codec = (AVCodecID)stream.iCodecId;
     dStream->uniqueId = stream.iPID;
     dStream->language = stream.strLanguage;
-    dStream->realtime = true;
+    dStream->realtime = false;

     m_newStreamMap[stream.iPID] = dStream;
   }
