diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
index a636814..0c9becd 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecAmlogic.cpp
@@ -106,8 +106,6 @@ bool CDVDVideoCodecAmlogic::Open(CDVDStreamInfo &hints, CDVDCodecOptions &option
     return false;
   }
 
-  m_opened = false;
-
   // allow only 1 instance here
   if (m_InstanceGuard.exchange(true))
   {
@@ -115,6 +113,8 @@ bool CDVDVideoCodecAmlogic::Open(CDVDStreamInfo &hints, CDVDCodecOptions &option
     return false;
   }
 
+  m_opened = false;
+
   m_hints = hints;
 
   CLog::Log(LOGDEBUG, "CDVDVideoCodecAmlogic::Opening: codec %d profile:%d extra_size:%d", m_hints.codec, hints.profile, hints.extrasize);
@@ -282,7 +282,6 @@ bool CDVDVideoCodecAmlogic::Open(CDVDStreamInfo &hints, CDVDCodecOptions &option
   CLog::Log(LOGINFO, "%s: Opened Amlogic Codec", __MODULE_NAME__);
   return true;
 FAIL:
-  m_InstanceGuard.exchange(false);
   Dispose();
   return false;
 }
@@ -294,9 +293,6 @@ void CDVDVideoCodecAmlogic::Dispose(void)
   if (m_Codec)
     m_Codec->CloseDecoder(), m_Codec = nullptr;
 
-  if (m_opened)
-    m_InstanceGuard.exchange(false);
-
   m_videobuffer.iFlags = 0;
 
   if (m_mpeg2_sequence)
@@ -309,6 +305,7 @@ void CDVDVideoCodecAmlogic::Dispose(void)
     delete m_bitparser, m_bitparser = NULL;
 
   m_opened = false;
+  m_InstanceGuard.exchange(false);
 }
 
 bool CDVDVideoCodecAmlogic::AddData(const DemuxPacket &packet)
