diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
index b0688e0..3e26532 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -1585,6 +1585,12 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
     }
   }
 
+  // set dynamic_buf_num_margin for hevc
+  if (hints.codec == AV_CODEC_ID_HEVC && hints.width <= 1280)
+    SysfsUtils::SetString("sys/module/amvdec_h265/parameters/dynamic_buf_num_margin", "16");
+  else
+    SysfsUtils::SetString("sys/module/amvdec_h265/parameters/dynamic_buf_num_margin", "8");
+
   // handle orientation
   am_private->video_rotation_degree = 0;
   if (hints.orientation == 90)
