diff --git a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
index ec445db..dcedbb2 100644
--- a/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
+++ b/xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp
@@ -1715,6 +1715,20 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
       am_private->gcodec.param  = (void*)EXTERNAL_PTS;
       if (m_hints.ptsinvalid)
         am_private->gcodec.param = (void*)(EXTERNAL_PTS | SYNC_OUTSIDE);
+      if (hints.extrasize > 24)
+        for (int i = 0; i < 24; i += 8)
+          CLog::Log(LOGDEBUG, LOGVIDEO, "%04x: %02x %02x %02x %02x %02x %02x %02x %02x ", i,
+             *((char *)hints.extradata + i + 0), *((char *)hints.extradata + i + 1), *((char *)hints.extradata + i + 2), *((char *)hints.extradata + i + 3),
+             *((char *)hints.extradata + i + 4), *((char *)hints.extradata + i + 5), *((char *)hints.extradata + i + 6), *((char *)hints.extradata + i + 7));
+      {
+        std::string s = "16";
+        if ((hints.extrasize < 17) || (*((char *)hints.extradata + 10) & 0x20))
+          s = "8";
+        else if ((*((char *)hints.extradata + 11) & 0x20))
+          s = "12";
+        SysfsUtils::SetString("/sys/module/amvdec_h265/parameters/dynamic_buf_num_margin", s);
+        CLog::Log(LOGDEBUG, LOGVIDEO,"CAMLCodec::OpenDecoder Set HEVC dynamic_buf_num_margin: %s", s.c_str());
+      }
       break;
     case VFORMAT_VP9:
       am_private->gcodec.format = VIDEO_DEC_FORMAT_VP9;
