diff -Naurp pvr.iptvsimple-c06f367/lib/rapidxml/rapidxml.hpp pvr.iptvsimple-c06f367-patch/lib/rapidxml/rapidxml.hpp
--- pvr.iptvsimple-c06f367/lib/rapidxml/rapidxml.hpp	2017-01-20 09:42:57.000000000 +0200
+++ pvr.iptvsimple-c06f367-patch/lib/rapidxml/rapidxml.hpp	2017-01-21 19:47:00.000000000 +0200
@@ -1596,7 +1596,23 @@ namespace rapidxml
                         {
 
                         // &amp; &apos;
-                        case Ch('a'): 
+                        case Ch('a'):
+                            // fix &amp;quot;
+                            if (src[2] == Ch('m') && src[3] == Ch('p') && src[4] == Ch(';') && src[5] == Ch('q') && src[6] == Ch('u') && src[7] == Ch('o') && src[8] == Ch('t') && src[9] == Ch(';'))
+                            {
+                                *dest = Ch('"');
+                                ++dest;
+                                src += 10;
+                                continue;
+                            }
+                            // fix &amp;apos;
+                            if (src[2] == Ch('m') && src[3] == Ch('p') && src[4] == Ch(';') && src[5] == Ch('a') && src[6] == Ch('p') && src[7] == Ch('o') && src[8] == Ch('s') && src[9] == Ch(';'))
+                            {
+                                *dest = Ch('\'');
+                                ++dest;
+                                src += 10;
+                                continue;
+                            }
                             if (src[2] == Ch('m') && src[3] == Ch('p') && src[4] == Ch(';'))
                             {
                                 *dest = Ch('&');
