--- a/bluetool/__init__.py
+++ b/bluetool/__init__.py
@@ -2,3 +2,4 @@
 from .blueserver import BluetoothServer
 
 __all__ = ['Bluetooth', 'BluetoothServer']
+from tcpbridge import TCPBridge, TCPBridgeError, TCPServer, SocketSink, FileSink
--- a/bluetool/blueserver.py
+++ b/bluetool/blueserver.py
@@ -29,6 +29,8 @@
 import socket
 
 try:
+    import pgi
+    pgi.install_as_gi()
     from gi.repository import GObject
 except ImportError:
     import gobject as GObject
