From 16e6e8424c365142af0a5020f15031e36abd07a0 Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Wed, 08 Jan 2020 22:58:43 +0100
Subject: [PATCH] build

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -412,6 +412,9 @@
 # ------------------------------------------------------------
 # Directory creation
 
+.PHONY: builddirs
+builddirs: $(OBJDIR) $(LIBDIR) $(BINDIR)
+
 $(OBJDIR) $(LIBDIR) $(BINDIR) $(DESTDIR)$(bindir) $(DESTDIR)$(libdir) $(DESTDIR)$(man1dir):
 	mkdir -p $@
 
