From 7273d940948298b35654da1bfb2ebba767d75d2a Mon Sep 17 00:00:00 2001
From: Andre Heider <a.heider@gmail.com>
Date: Sat, 14 Dec 2019 13:15:23 +0100
Subject: [PATCH] crossstrip

diff --git a/Makefile b/Makefile
--- a/Makefile
+++ b/Makefile
@@ -43,7 +43,7 @@
 man1dir=$(mandir)/man1
 manext=.1
 
-INSTALL=install
+INSTALL=install --strip-program=$(CROSS_COMPILE)strip
 INSTALL_PROGRAM=$(INSTALL) -m 0555 -s
 INSTALL_LIB=$(INSTALL) -m 0444 -s
 INSTALL_DATA=$(INSTALL) -m 0444
