--- a/src/gallium/targets/dri/meson.build	2026-07-16 22:43:10.635701052 +0000
+++ b/src/gallium/targets/dri/meson.build	2026-07-16 22:43:19.569121376 +0000
@@ -47,7 +47,7 @@
     inc_gallium_winsys, include_directories('../../frontends/dri'),
   ],
   gnu_symbol_visibility : 'hidden',
-  link_args : [ld_args_build_id, ld_args_gc_sections, gallium_dri_ld_args],
+  link_args : [ld_args_build_id, ld_args_gc_sections, gallium_dri_ld_args, '-lgcc'],
   link_depends : gallium_dri_link_depends,
   link_with : [
     libmesa, libgalliumvl,
