From ee19e2675554c02733fd9af20617724a2cf038dc Mon Sep 17 00:00:00 2001
From: Rudi Heitbaum <rudi@heitbaum.com>
Date: Tue, 12 Jul 2022 13:56:23 +0000
Subject: [PATCH] include ctime in timer

https://bugs.gentoo.org/851837

---
diff --git a/src/timer.cpp b/src/timer.cpp
index f0a98b3..0a66a93 100644
--- a/src/timer.cpp
+++ b/src/timer.cpp
@@ -27,6 +27,7 @@
 #include "cxxtools/clock.h"
 #include "cxxtools/selector.h"
 #include "cxxtools/datetime.h"
+#include <ctime>
 #include <stdexcept>
 
 namespace cxxtools
-- 
2.53.0

