From aa384798e76ab79b642fef43d49b5c09a83cc454 Mon Sep 17 00:00:00 2001
From: SupervisedThinking <supervisedthinking@gmail.com>
Date: Thu, 15 Dec 2022 14:35:02 +0100
Subject: [PATCH] build static spirv tools

diff --git a/External/CMakeLists.txt b/External/CMakeLists.txt
--- a/External/CMakeLists.txt
+++ b/External/CMakeLists.txt
@@ -71,7 +71,8 @@ endif()
 if(ENABLE_OPT AND NOT TARGET SPIRV-Tools-opt)
     if(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/spirv-tools)
         set(SPIRV_SKIP_TESTS ON CACHE BOOL "Skip building SPIRV-Tools tests")
-        add_subdirectory(spirv-tools)
+        set(BUILD_SHARED_LIBS OFF CACHE BOOL "Disable build shared libs" FORCE)
+        add_subdirectory(spirv-tools EXCLUDE_FROM_ALL)
     endif()
 endif()
 
