--- a/External/CMakeLists.txt
+++ b/External/CMakeLists.txt
@@ -71,7 +71,8 @@ endif()
 if(ENABLE_OPT AND NOT TARGET SPIRV-Tools-opt)
     if(IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/spirv-tools)
         set(SPIRV_SKIP_TESTS ON CACHE BOOL "Skip building SPIRV-Tools tests")
-        add_subdirectory(spirv-tools)
+        set(BUILD_SHARED_LIBS OFF CACHE BOOL "Disable build shared libs" FORCE)
+        add_subdirectory(spirv-tools EXCLUDE_FROM_ALL)
     endif()
 endif()
 
