From eff30ed6608eceecfa8c589c151e3eefcb46a0d1 Mon Sep 17 00:00:00 2001
From: Adam Green <greena88@gmail.com>
Date: Wed, 13 Dec 2017 00:58:43 +0000
Subject: [PATCH] firmware path

diff --git a/hciattach_rtk.c b/hciattach_rtk.c
--- a/hciattach_rtk.c
+++ b/hciattach_rtk.c
@@ -1419,7 +1419,7 @@
 	int ret = 0;
 	struct stat st;
 
-	ret = sprintf(firmware_file_name, FIRMWARE_DIRECTORY"rtlbt_fw");
+	ret = sprintf(firmware_file_name, FIRMWARE_DIRECTORY"rtl8723b_fw");
 
 	return firmware_file_name;
 }
@@ -1504,7 +1504,7 @@
 	int ret = 0;
 	int i = 0;
 
-	ret = sprintf(bt_config_file_name, BT_CONFIG_DIRECTORY"rtlbt_config"); 
+	ret = sprintf(bt_config_file_name, BT_CONFIG_DIRECTORY"rtl8723b_config");
 	if (stat(bt_config_file_name, &st) < 0) {
 		RS_ERR("can't access bt config file:%s, errno:%d\n", bt_config_file_name, errno);
 		return -1;
