From 60b222a7b3007519239f96837482418a632ab3ad Mon Sep 17 00:00:00 2001
From: Matthias Reichl <hias@horus.com>
Date: Wed, 06 Aug 2025 12:18:15 +0200
Subject: [PATCH] display null prompt

---
 display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/display.c b/display.c
--- a/display.c
+++ b/display.c
@@ -784,5 +784,5 @@
 /* Useful shorthand used by rl_redisplay, update_line, rl_move_cursor_relative */
 #define INVIS_FIRST()	(local_prompt_invis_chars[0])
-#define WRAP_OFFSET(line, offset)  ((line <= prompt_last_screen_line) ? local_prompt_invis_chars[line] : 0)
+#define WRAP_OFFSET(line, offset)  ((line <= prompt_last_screen_line && local_prompt_invis_chars) ? local_prompt_invis_chars[line] : 0)

 #define W_OFFSET(line, offset) ((line) == 0 ? offset : 0)
--
2.47.1
