diff -Naur a/systemd/rpcbind.service.in b/systemd/rpcbind.service.in
--- a/systemd/rpcbind.service.in	2016-12-20 03:18:51.839794796 +0000
+++ b/systemd/rpcbind.service.in	2016-12-20 03:21:30.476854474 +0000
@@ -8,6 +8,7 @@
 # rpcbind.socket, no matter how this unit is started.
 Requires=rpcbind.socket
 Wants=rpcbind.target
+After=rpcbind.socket systemd-tmpfiles-setup.service
 
 [Service]
 Type=notify
