GLideN64 don't build static

--- a/src/CMakeLists.txt1	2015-10-19 16:43:56.000000000 +0200
+++ b/src/CMakeLists.txt	2019-02-28 19:25:32.730341450 +0100
@@ -207,7 +207,7 @@ if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "
   if (G++_VERSION VERSION_LESS 4.8)
 	message(SEND_ERROR "You need at least G++ 4.8 to compile GLideN64!")
   endif()
-  SET(GCC_CPP11_COMPILE_FLAGS "-std=c++0x -static-libgcc -static-libstdc++")
+  SET(GCC_CPP11_COMPILE_FLAGS "-std=c++0x")
   SET( CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${GCC_CPP11_COMPILE_FLAGS}" )
   SET(GCC_STATIC_LINK_FLAGS "-static-libgcc -static-libstdc++")
   SET( CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} ${GCC_STATIC_LINK_FLAGS}" )
