--- a/Makefile
+++ b/Makefile
@@ -501,7 +501,7 @@
 	ARM_FLOAT_ABI_HARD = 0
 	FORCE_GLES = 1
 	SINGLE_PREC_FLAGS = 1
-	CPUFLAGS += -DTARGET_LINUX_ARMv8 -frename-registers
+	CPUFLAGS += -DTARGET_LINUX_ARMv8 -frename-registers -mno-outline-atomics
 	CFLAGS += -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53 $(CPUFLAGS)
 	CXXFLAGS += -mcpu=cortex-a73 -mtune=cortex-a73.cortex-a53 $(CPUFLAGS)
 	ASFLAGS += $(CFLAGS) -c -frename-registers -fno-strict-aliasing -ffast-math -ftree-vectorize
