--- a/Makefile
+++ b/Makefile
@@ -19,6 +19,7 @@
 HAVE_CLANG    ?= 0
 HAVE_CDROM    := 0
 THREADED_RENDERING_DEFAULT := 0
+SET_OPTIM     := -O3
 
 
 TARGET_NAME   := flycast
@@ -889,7 +890,7 @@
 	ifneq (,$(findstring msvc,$(platform)))
 		OPTFLAGS       := -O2
 	else ifneq ($(platform), classic_armv7_a7)
-		OPTFLAGS       := -O3
+		OPTFLAGS       := $(SET_OPTIM)
 	endif
 
 	CORE_DEFINES   += -DNDEBUG
