From 4df24cf8ce5a55eefd3ca6065fc65e8b374c9ff7 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Tue, 2 Jun 2015 17:52:40 +0100
Subject: [PATCH] Do not present CEC toasts unless there is an error or library
 mismatch

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/peripherals/devices/PeripheralCecAdapter.cpp | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
index 802dd0779e..83065c93f6 100644
--- a/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
+++ b/xbmc/peripherals/devices/PeripheralCecAdapter.cpp
@@ -277,9 +277,6 @@ bool CPeripheralCecAdapter::OpenConnection(void)
   // open the CEC adapter
   CLog::Log(LOGDEBUG, "%s - opening a connection to the CEC adapter: %s", __FUNCTION__, m_strComPort.c_str());
 
-  // scanning the CEC bus takes about 5 seconds, so display a notification to inform users that we're busy
-  std::string strMessage = StringUtils::Format(g_localizeStrings.Get(21336).c_str(), g_localizeStrings.Get(36000).c_str());
-  CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, g_localizeStrings.Get(36000), strMessage);
 
   bool bConnectionFailedDisplayed(false);
 
@@ -696,7 +693,6 @@ void CPeripheralCecAdapter::CecAlert(void *cbParam, const libcec_alert alert, co
     break;
   case CEC_ALERT_CONNECTION_LOST:
     bReopenConnection = true;
-    iAlertString = 36030;
     break;
 #if defined(CEC_ALERT_PERMISSION_ERROR)
   case CEC_ALERT_PERMISSION_ERROR:
@@ -1566,12 +1562,7 @@ bool CPeripheralCecAdapterUpdateThread::SetInitialConfiguration(void)
   std::string strAmpName = UpdateAudioSystemStatus();
   if (!strAmpName.empty())
     strNotification += StringUtils::Format("- %s", strAmpName.c_str());
-
   m_adapter->m_bIsReady = true;
-
-  // and let the gui know that we're done
-  CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, g_localizeStrings.Get(36000), strNotification);
-
   CSingleLock lock(m_critSection);
   m_bIsUpdating = false;
   return true;
@@ -1617,8 +1608,6 @@ void CPeripheralCecAdapterUpdateThread::Process(void)
         UpdateAudioSystemStatus();
       }
 
-      CGUIDialogKaiToast::QueueNotification(CGUIDialogKaiToast::Info, g_localizeStrings.Get(36000), g_localizeStrings.Get(bConfigSet ? 36023 : 36024));
-
       {
         CSingleLock lock(m_critSection);
         if ((bUpdate = m_bNextConfigurationScheduled) == true)

--
2.17.1
