--- a/OpenVFDService.c
+++ b/OpenVFDService.c
@@ -18,7 +18,7 @@
 #include <signal.h>
 
 #include <time.h>
-#include "driver/open_vfd_drv.h"
+#include "driver/openvfd_drv.h"
 
 #define UNUSED(x)	(void*)(x)
 #define DRV_NAME	"/dev/" DEV_NAME
--- a/Makefile
+++ b/Makefile
@@ -2,5 +2,5 @@
 	obj-m += driver/openvfd.o
 	$(MAKE) modules
 
-FD628Service: OpenVFDService.c
+OpenVFDService: OpenVFDService.c
 	$(CC) $(CFLAGS) -Wall -w -o $@ $^ -lm -lpthread
